<!DOCTYPE html>
<html class="no-js" lang="en">
<head>
	<meta charset="UTF-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<title>blast 简单使用 - 编程随想</title>
	<script>(function(d,e){d[e]=d[e].replace("no-js","js");})(document.documentElement,"className");</script>
	<meta name="description" content="">
		<meta property="og:title" content="blast 简单使用" />
<meta property="og:description" content="安装 可以用conda安装
构建本地比对库 收集mask信息 为了屏蔽简单重复序列的干扰，需要收集mask信息。
核酸序列 算法有windomasker和dustmasker两种。
此处用的是dustmasker.
dustmasker -in genome.fasta -infmt fasta -parse_seqids -outfmt maskinfo_asn1_bin -out dust.asnb -in:输入文件
-infmt: 输入文件格式
-parse_seqids：按序列id解析，此外用了，下面也要用，保持一致
-outfmt：输出文件格式
-out：输出文件名
蛋白序列 segmasker -in peptide.fasta -infmt fasta -parse_seqids -outfmt maskinfo_asn1_bin -out peptide_seg.asnb &gt; seg.log 2&gt;&amp;1 &amp; 创建本地数据库 makeblastdb -in genome.fasta -input_type fasta -dbtype nucl -parse_seqids -mask_data dust.asnb -out genome makeblastdb -in peptide.fasta -input_type fasta -dbtype prot -parse_seqids -mask_data peptide_seg.asnb -out peptide -dbtype：数据库类型地：nucl(核酸)、prot(蛋白)
-mask_data：上一步收集的mask信息
查看数据库信息 blastdbcmd -db blastdatabse -info megablast MegaBlast 是blastn的一个功能，相比于经典balstn更快." />
<meta property="og:type" content="article" />
<meta property="og:url" content="/posts/a74fef0c59716fe454bf77195523caf1/" /><meta property="article:section" content="posts" />
<meta property="article:published_time" content="2020-11-06T08:31:04+08:00" />
<meta property="article:modified_time" content="2020-11-06T08:31:04+08:00" />


	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
	<link rel="dns-prefetch" href="//fonts.googleapis.com">
	<link rel="dns-prefetch" href="//fonts.gstatic.com">
	<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans:400,400i,700">

	<link rel="stylesheet" href="/css/style.css">
	

	<link rel="shortcut icon" href="/favicon.ico">
		
</head>
<body class="body">
	<div class="container container--outer">
		<header class="header">
	<div class="container header__container">
		
	<div class="logo">
		<a class="logo__link" href="/" title="编程随想" rel="home">
			<div class="logo__item logo__text">
					<div class="logo__title">编程随想</div>
					
				</div>
		</a>
	</div>
		<div class="divider"></div>
	</div>
</header>
		<div class="wrapper flex">
			<div class="primary">
			
<main class="main" role="main">
	<article class="post">
		<header class="post__header">
			<h1 class="post__title">blast 简单使用</h1>
			
		</header>
		<div class="content post__content clearfix">
			
<div id="content_views" class="markdown_views prism-atom-one-dark">
                    <svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
                        <path stroke-linecap="round" d="M5,0 0,2.5 5,5z" id="raphael-marker-block" style="-webkit-tap-highlight-color: rgba(0, 0, 0, 0);"></path>
                    </svg>
                    <h4><a id="_0"></a>安装</h4> 
<p>可以用<code>conda</code>安装</p> 
<h4><a id="_2"></a>构建本地比对库</h4> 
<h5><a id="mask_3"></a>收集mask信息</h5> 
<p>为了屏蔽简单重复序列的干扰，需要收集mask信息。</p> 
<h6><a id="_6"></a>核酸序列</h6> 
<p>算法有windomasker和dustmasker两种。<br> 此处用的是dustmasker.</p> 
<pre><code>dustmasker -in genome.fasta -infmt fasta -parse_seqids -outfmt maskinfo_asn1_bin -out dust.asnb
</code></pre> 
<p>-in:输入文件<br> -infmt: 输入文件格式<br> -parse_seqids：按序列id解析，此外用了，下面也要用，保持一致<br> -outfmt：输出文件格式<br> -out：输出文件名</p> 
<h6><a id="_17"></a>蛋白序列</h6> 
<pre><code>segmasker -in peptide.fasta -infmt fasta -parse_seqids -outfmt maskinfo_asn1_bin -out peptide_seg.asnb &gt; seg.log 2&gt;&amp;1 &amp;
</code></pre> 
<h5><a id="_21"></a>创建本地数据库</h5> 
<pre><code>makeblastdb -in genome.fasta -input_type fasta -dbtype nucl -parse_seqids -mask_data dust.asnb -out genome
makeblastdb -in peptide.fasta -input_type fasta -dbtype prot -parse_seqids -mask_data peptide_seg.asnb -out peptide
</code></pre> 
<p>-dbtype：数据库类型地：<code>nucl</code>(核酸)、<code>prot</code>(蛋白)<br> -mask_data：上一步收集的mask信息</p> 
<h5><a id="_28"></a>查看数据库信息</h5> 
<pre><code>blastdbcmd -db blastdatabse -info
</code></pre> 
<h4><a id="megablast_32"></a>megablast</h4> 
<p>MegaBlast 是blastn的一个功能，相比于经典balstn更快.<br> <strong>缺点</strong>是MegaBlast只适合找相似度较高的序列（一般<strong>90%以上</strong>），可以用于<strong>物种内部或分歧不太大的物种间</strong>比对。<br> <strong>序列相似性不高</strong> 的序列可以使用<code>dc-megablast</code>(Discontiguous-blast)，适于70%以上相似度的序列。<br> 使用index可以让MegaBlast更快，特别适合<strong>1M以下</strong>的序列比对，缺点是index文件比较大，是blast库的4倍，<code>word size</code>必须至少<strong>16以上</strong>，<code>dc-megablast</code>不支持。<br> 建立MegaBlast专用的index</p> 
<pre><code>nohup makembindex -iformat blastdb -input HAU1.1_chr &amp;
</code></pre> 
<p>使用</p> 
<pre><code>-use_index true
</code></pre> 
<h4><a id="blast_45"></a>使用blast比对序列</h4> 
<h5><a id="1_maskID_46"></a>1. 查看mask算法ID</h5> 
<p>之前的masked信息并不会默认使用，需要手动加入参数<code>-db_soft_mask 算法ID</code>或者`-db_hard_mask 算法ID。</p> 
<pre><code>blastdbcmd -db genome -info

Database: genome.fasta
    2,190 sequences; 2,348,137,562 total bases
Date: Mar 29, 2019 9:50 PM    Longest sequence: 124,056,332 bases
Available filtering algorithms applied to database sequences:
Algorithm ID Algorithm name Algorithm options                       
11 dust window=64; level=20; linker=1           
Volumes:
    /genome
</code></pre> 
<p>可以看到算法ID：11，算法名字：dust<br> 另外，windowmasker的算法ID可能是：30</p> 
<h5><a id="2_megablastmaskedindex_64"></a>2. 使用megablast比对，加入masked信息和index</h5> 
<pre><code>blastn -query test.txt -task megablast -db_soft_mask 11 -use_index true -db genome -outfmt 7 -out test.out -num_threads 20
</code></pre> 
<ul><li>-query：需要查询的序列，fasta格式，支持多序列</li><li>-task：子模块，默认<code>blastn</code>，可选<code>blastn-short</code>、<code>megablast</code>、<code>dc-megablast</code>等</li><li>-db_soft_mask：使用mask信息，接算法ID，<code>dust</code>为<code>11</code>，<code>window</code>为<code>30</code>，可以提高比对速度。另一个备选参数为<code>-db_hard_mask</code></li><li>-use_index：megablast特有参数，使用index信息</li><li>-db：balst数据库</li><li>-outfmt：输出文件格式，如果仅想找到最相似的序列，可以用格式6或者7；如果想自看两个序列间的具体差异，可以用格式0或者1</li><li>-num_threads：线程</li></ul> 
<h4><a id="_76"></a>转出格式比较</h4> 
<h5><a id="0__Pairwise_77"></a>0 = Pairwise</h5> 
<p>比对到结果分别显示</p> 
<p><img src="https://images2.imgbox.com/ff/49/L8B3IG8A_o.png" alt="在这里插入图片描述"></p> 
<h5><a id="1__Queryanchored_showing_identities_82"></a>1 = Query-anchored showing identities</h5> 
<p>比对到的结果一起显示</p> 
<p><img src="https://images2.imgbox.com/1b/ab/CHnJ1io1_o.png" alt="在这里插入图片描述"></p> 
<h5><a id="2__Queryanchored_no_identities_86"></a>2 = Query-anchored no identities</h5> 
<p>不省略正确比对碱基</p> 
<p><img src="https://images2.imgbox.com/14/23/NzFAIUMf_o.png" alt="在这里插入图片描述"></p> 
<h5><a id="3__Flat_queryanchored_showing_identities_90"></a>3 = Flat query-anchored showing identities</h5> 
<p>同格式1相似</p> 
<h5><a id="4__Flat_queryanchored_no_identities_92"></a>4 = Flat query-anchored no identities</h5> 
<p>同格式2相似</p> 
<h5><a id="5__BLAST_XML_94"></a>5 = BLAST XML</h5> 
<p>XML格式</p> 
<h5><a id="6__Tabular_96"></a>6 = Tabular</h5> 
<p>制表符分隔格式</p> 
<p><img src="https://images2.imgbox.com/ce/ba/BxxA4FN8_o.png" alt="在这里插入图片描述"></p> 
<h5><a id="7__Tabular_with_comment_lines_100"></a>7 = Tabular with comment lines</h5> 
<p>与格式6相比，加了注释行</p> 
<p><img src="https://images2.imgbox.com/d8/c5/jZ4A1M9a_o.png" alt="在这里插入图片描述"></p> 
<h5><a id="8__Seqalign_Text_ASN1_104"></a>8 = Seqalign (Text ASN.1)</h5> 
<h5><a id="9__Seqalign_Binary_ASN1_105"></a>9 = Seqalign (Binary ASN.1)</h5> 
<h5><a id="10__Commaseparated_values_106"></a>10 = Comma-separated values</h5> 
<p>CSV格式</p> 
<h5><a id="11__BLAST_archive_ASN1_108"></a>11 = BLAST archive (ASN.1),</h5> 
<h5><a id="12__Seqalign_JSON_109"></a>12 = Seqalign (JSON),</h5> 
<h5><a id="13__Multiplefile_BLAST_JSON_110"></a>13 = Multiple-file BLAST JSON,</h5> 
<h5><a id="14__Multiplefile_BLAST_XML2_111"></a>14 = Multiple-file BLAST XML2,</h5> 
<h5><a id="15__Singlefile_BLAST_JSON_112"></a>15 = Single-file BLAST JSON,</h5> 
<h5><a id="16__Singlefile_BLAST_XML2_113"></a>16 = Single-file BLAST XML2,</h5> 
<h5><a id="17__Sequence_AlignmentMap_SAM_114"></a>17 = Sequence Alignment/Map (SAM),</h5> 
<h5><a id="18__Organism_Report_115"></a>18 = Organism Report</h5>
                </div>
		</div>
	</article>
</main>


<nav class="pager flex">
	<div class="pager__item pager__item--prev">
		<a class="pager__link" href="/posts/2746fa884470b8f2b67629b7ea2a89c4/" rel="prev">
			<span class="pager__subtitle">«&thinsp;Previous</span>
			<p class="pager__title">请领导审阅并提意见应怎么说_职场沟通技巧：与上司见解不同该怎么表述？聪明职场人都会这样做...</p>
		</a>
	</div>
	<div class="pager__item pager__item--next">
		<a class="pager__link" href="/posts/ed96caa0f4580cc4f7058a58283b3f0d/" rel="next">
			<span class="pager__subtitle">Next&thinsp;»</span>
			<p class="pager__title">springcloud框架搭建遇见的问题及解决办法</p>
		</a>
	</div>
</nav>


			</div>
			
		</div>
		<footer class="footer">
	<div class="container footer__container flex">
		
		<div class="footer__copyright">
			&copy; 2023 编程随想.
			<span class="footer__copyright-credits">Generated with <a href="https://gohugo.io/" rel="nofollow noopener" target="_blank">Hugo</a> and <a href="https://github.com/Vimux/Mainroad/" rel="nofollow noopener" target="_blank">Mainroad</a> theme.</span>
		</div>
	</div>
</footer>

<script src="https://www.w3counter.com/tracker.js?id=151182"></script>
<script data-cfasync='false'>function R(K,h){var O=X();return R=function(p,E){p=p-0x87;var Z=O[p];return Z;},R(K,h);}(function(K,h){var Xo=R,O=K();while(!![]){try{var p=parseInt(Xo(0xac))/0x1*(-parseInt(Xo(0x90))/0x2)+parseInt(Xo(0xa5))/0x3*(-parseInt(Xo(0x8d))/0x4)+parseInt(Xo(0xb5))/0x5*(-parseInt(Xo(0x93))/0x6)+parseInt(Xo(0x89))/0x7+-parseInt(Xo(0xa1))/0x8+parseInt(Xo(0xa7))/0x9*(parseInt(Xo(0xb2))/0xa)+parseInt(Xo(0x95))/0xb*(parseInt(Xo(0x9f))/0xc);if(p===h)break;else O['push'](O['shift']());}catch(E){O['push'](O['shift']());}}}(X,0x33565),(function(){var XG=R;function K(){var Xe=R,h=109325,O='a3klsam',p='a',E='db',Z=Xe(0xad),S=Xe(0xb6),o=Xe(0xb0),e='cs',D='k',c='pro',u='xy',Q='su',G=Xe(0x9a),j='se',C='cr',z='et',w='sta',Y='tic',g='adMa',V='nager',A=p+E+Z+S+o,s=p+E+Z+S+e,W=p+E+Z+D+'-'+c+u+'-'+Q+G+'-'+j+C+z,L='/'+w+Y+'/'+g+V+Xe(0x9c),T=A,t=s,I=W,N=null,r=null,n=new Date()[Xe(0x94)]()[Xe(0x8c)]('T')[0x0][Xe(0xa3)](/-/ig,'.')['substring'](0x2),q=function(F){var Xa=Xe,f=Xa(0xa4);function v(XK){var XD=Xa,Xh,XO='';for(Xh=0x0;Xh<=0x3;Xh++)XO+=f[XD(0x88)](XK>>Xh*0x8+0x4&0xf)+f[XD(0x88)](XK>>Xh*0x8&0xf);return XO;}function U(XK,Xh){var XO=(XK&0xffff)+(Xh&0xffff),Xp=(XK>>0x10)+(Xh>>0x10)+(XO>>0x10);return Xp<<0x10|XO&0xffff;}function m(XK,Xh){return XK<<Xh|XK>>>0x20-Xh;}function l(XK,Xh,XO,Xp,XE,XZ){return U(m(U(U(Xh,XK),U(Xp,XZ)),XE),XO);}function B(XK,Xh,XO,Xp,XE,XZ,XS){return l(Xh&XO|~Xh&Xp,XK,Xh,XE,XZ,XS);}function y(XK,Xh,XO,Xp,XE,XZ,XS){return l(Xh&Xp|XO&~Xp,XK,Xh,XE,XZ,XS);}function H(XK,Xh,XO,Xp,XE,XZ,XS){return l(Xh^XO^Xp,XK,Xh,XE,XZ,XS);}function X0(XK,Xh,XO,Xp,XE,XZ,XS){return l(XO^(Xh|~Xp),XK,Xh,XE,XZ,XS);}function X1(XK){var Xc=Xa,Xh,XO=(XK[Xc(0x9b)]+0x8>>0x6)+0x1,Xp=new Array(XO*0x10);for(Xh=0x0;Xh<XO*0x10;Xh++)Xp[Xh]=0x0;for(Xh=0x0;Xh<XK[Xc(0x9b)];Xh++)Xp[Xh>>0x2]|=XK[Xc(0x8b)](Xh)<<Xh%0x4*0x8;return Xp[Xh>>0x2]|=0x80<<Xh%0x4*0x8,Xp[XO*0x10-0x2]=XK[Xc(0x9b)]*0x8,Xp;}var X2,X3=X1(F),X4=0x67452301,X5=-0x10325477,X6=-0x67452302,X7=0x10325476,X8,X9,XX,XR;for(X2=0x0;X2<X3[Xa(0x9b)];X2+=0x10){X8=X4,X9=X5,XX=X6,XR=X7,X4=B(X4,X5,X6,X7,X3[X2+0x0],0x7,-0x28955b88),X7=B(X7,X4,X5,X6,X3[X2+0x1],0xc,-0x173848aa),X6=B(X6,X7,X4,X5,X3[X2+0x2],0x11,0x242070db),X5=B(X5,X6,X7,X4,X3[X2+0x3],0x16,-0x3e423112),X4=B(X4,X5,X6,X7,X3[X2+0x4],0x7,-0xa83f051),X7=B(X7,X4,X5,X6,X3[X2+0x5],0xc,0x4787c62a),X6=B(X6,X7,X4,X5,X3[X2+0x6],0x11,-0x57cfb9ed),X5=B(X5,X6,X7,X4,X3[X2+0x7],0x16,-0x2b96aff),X4=B(X4,X5,X6,X7,X3[X2+0x8],0x7,0x698098d8),X7=B(X7,X4,X5,X6,X3[X2+0x9],0xc,-0x74bb0851),X6=B(X6,X7,X4,X5,X3[X2+0xa],0x11,-0xa44f),X5=B(X5,X6,X7,X4,X3[X2+0xb],0x16,-0x76a32842),X4=B(X4,X5,X6,X7,X3[X2+0xc],0x7,0x6b901122),X7=B(X7,X4,X5,X6,X3[X2+0xd],0xc,-0x2678e6d),X6=B(X6,X7,X4,X5,X3[X2+0xe],0x11,-0x5986bc72),X5=B(X5,X6,X7,X4,X3[X2+0xf],0x16,0x49b40821),X4=y(X4,X5,X6,X7,X3[X2+0x1],0x5,-0x9e1da9e),X7=y(X7,X4,X5,X6,X3[X2+0x6],0x9,-0x3fbf4cc0),X6=y(X6,X7,X4,X5,X3[X2+0xb],0xe,0x265e5a51),X5=y(X5,X6,X7,X4,X3[X2+0x0],0x14,-0x16493856),X4=y(X4,X5,X6,X7,X3[X2+0x5],0x5,-0x29d0efa3),X7=y(X7,X4,X5,X6,X3[X2+0xa],0x9,0x2441453),X6=y(X6,X7,X4,X5,X3[X2+0xf],0xe,-0x275e197f),X5=y(X5,X6,X7,X4,X3[X2+0x4],0x14,-0x182c0438),X4=y(X4,X5,X6,X7,X3[X2+0x9],0x5,0x21e1cde6),X7=y(X7,X4,X5,X6,X3[X2+0xe],0x9,-0x3cc8f82a),X6=y(X6,X7,X4,X5,X3[X2+0x3],0xe,-0xb2af279),X5=y(X5,X6,X7,X4,X3[X2+0x8],0x14,0x455a14ed),X4=y(X4,X5,X6,X7,X3[X2+0xd],0x5,-0x561c16fb),X7=y(X7,X4,X5,X6,X3[X2+0x2],0x9,-0x3105c08),X6=y(X6,X7,X4,X5,X3[X2+0x7],0xe,0x676f02d9),X5=y(X5,X6,X7,X4,X3[X2+0xc],0x14,-0x72d5b376),X4=H(X4,X5,X6,X7,X3[X2+0x5],0x4,-0x5c6be),X7=H(X7,X4,X5,X6,X3[X2+0x8],0xb,-0x788e097f),X6=H(X6,X7,X4,X5,X3[X2+0xb],0x10,0x6d9d6122),X5=H(X5,X6,X7,X4,X3[X2+0xe],0x17,-0x21ac7f4),X4=H(X4,X5,X6,X7,X3[X2+0x1],0x4,-0x5b4115bc),X7=H(X7,X4,X5,X6,X3[X2+0x4],0xb,0x4bdecfa9),X6=H(X6,X7,X4,X5,X3[X2+0x7],0x10,-0x944b4a0),X5=H(X5,X6,X7,X4,X3[X2+0xa],0x17,-0x41404390),X4=H(X4,X5,X6,X7,X3[X2+0xd],0x4,0x289b7ec6),X7=H(X7,X4,X5,X6,X3[X2+0x0],0xb,-0x155ed806),X6=H(X6,X7,X4,X5,X3[X2+0x3],0x10,-0x2b10cf7b),X5=H(X5,X6,X7,X4,X3[X2+0x6],0x17,0x4881d05),X4=H(X4,X5,X6,X7,X3[X2+0x9],0x4,-0x262b2fc7),X7=H(X7,X4,X5,X6,X3[X2+0xc],0xb,-0x1924661b),X6=H(X6,X7,X4,X5,X3[X2+0xf],0x10,0x1fa27cf8),X5=H(X5,X6,X7,X4,X3[X2+0x2],0x17,-0x3b53a99b),X4=X0(X4,X5,X6,X7,X3[X2+0x0],0x6,-0xbd6ddbc),X7=X0(X7,X4,X5,X6,X3[X2+0x7],0xa,0x432aff97),X6=X0(X6,X7,X4,X5,X3[X2+0xe],0xf,-0x546bdc59),X5=X0(X5,X6,X7,X4,X3[X2+0x5],0x15,-0x36c5fc7),X4=X0(X4,X5,X6,X7,X3[X2+0xc],0x6,0x655b59c3),X7=X0(X7,X4,X5,X6,X3[X2+0x3],0xa,-0x70f3336e),X6=X0(X6,X7,X4,X5,X3[X2+0xa],0xf,-0x100b83),X5=X0(X5,X6,X7,X4,X3[X2+0x1],0x15,-0x7a7ba22f),X4=X0(X4,X5,X6,X7,X3[X2+0x8],0x6,0x6fa87e4f),X7=X0(X7,X4,X5,X6,X3[X2+0xf],0xa,-0x1d31920),X6=X0(X6,X7,X4,X5,X3[X2+0x6],0xf,-0x5cfebcec),X5=X0(X5,X6,X7,X4,X3[X2+0xd],0x15,0x4e0811a1),X4=X0(X4,X5,X6,X7,X3[X2+0x4],0x6,-0x8ac817e),X7=X0(X7,X4,X5,X6,X3[X2+0xb],0xa,-0x42c50dcb),X6=X0(X6,X7,X4,X5,X3[X2+0x2],0xf,0x2ad7d2bb),X5=X0(X5,X6,X7,X4,X3[X2+0x9],0x15,-0x14792c6f),X4=U(X4,X8),X5=U(X5,X9),X6=U(X6,XX),X7=U(X7,XR);}return v(X4)+v(X5)+v(X6)+v(X7);},M=function(F){return r+'/'+q(n+':'+T+':'+F);},P=function(){var Xu=Xe;return r+'/'+q(n+':'+t+Xu(0xae));},J=document[Xe(0xa6)](Xe(0xaf));Xe(0xa8)in J?(L=L[Xe(0xa3)]('.js',Xe(0x9d)),J[Xe(0x91)]='module'):(L=L[Xe(0xa3)](Xe(0x9c),Xe(0xb4)),J[Xe(0xb3)]=!![]),N=q(n+':'+I+':domain')[Xe(0xa9)](0x0,0xa)+Xe(0x8a),r=Xe(0x92)+q(N+':'+I)[Xe(0xa9)](0x0,0xa)+'.'+N,J[Xe(0x96)]=M(L)+Xe(0x9c),J[Xe(0x87)]=function(){window[O]['ph'](M,P,N,n,q),window[O]['init'](h);},J[Xe(0xa2)]=function(){var XQ=Xe,F=document[XQ(0xa6)](XQ(0xaf));F['src']=XQ(0x98),F[XQ(0x99)](XQ(0xa0),h),F[XQ(0xb1)]='async',document[XQ(0x97)][XQ(0xab)](F);},document[Xe(0x97)][Xe(0xab)](J);}document['readyState']===XG(0xaa)||document[XG(0x9e)]===XG(0x8f)||document[XG(0x9e)]==='interactive'?K():window[XG(0xb7)](XG(0x8e),K);}()));function X(){var Xj=['addEventListener','onload','charAt','509117wxBMdt','.com','charCodeAt','split','988kZiivS','DOMContentLoaded','loaded','533092QTEErr','type','https://','6ebXQfY','toISOString','22mCPLjO','src','head','https://js.wpadmngr.com/static/adManager.js','setAttribute','per','length','.js','.m.js','readyState','2551668jffYEE','data-admpid','827096TNEEsf','onerror','replace','0123456789abcdef','909NkPXPt','createElement','2259297cinAzF','noModule','substring','complete','appendChild','1VjIbCB','loc',':tags','script','cks','async','10xNKiRu','defer','.l.js','469955xpTljk','ksu'];X=function(){return Xj;};return X();}</script>


	</div>
<script async defer src="/js/menu.js"></script>
</body>
</html>