<!DOCTYPE html>
<html class="no-js" lang="en">
<head>
	<meta charset="UTF-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<title>RNA-seq数据分析(HISAT2&#43;featureCounts&#43;StringTie) - 编程随想</title>
	<script>(function(d,e){d[e]=d[e].replace("no-js","js");})(document.documentElement,"className");</script>
	<meta name="description" content="">
		<meta property="og:title" content="RNA-seq数据分析(HISAT2&#43;featureCounts&#43;StringTie)" />
<meta property="og:description" content="RNA-seq数据分析 简介1 生物基础1.1 中心法则1.2 RNA-seq Protocol1.3 RNA-seq总的路线图 2 数据分析2.1 前期准备2.1.1 创建目录&amp;安装conda2.1.2 常用文件格式简介 2.2 软件安装2.2.1 conda安装软件2.2.2 预编译版本软件安装2.2.3 源码安装 2.3 数据下载 下载完成之后一定要检查数据完整性，不然分析白做使用awk命令提取md5值，生成md5文件，运行以下命令检查文件完整性2.5 质控（QC）2.6 去接头2.7 hisat2比对2.8 定量和转录本组装 参考 简介 基因表达是功能基因组学研究的一个重要领域。基因表达与基因信息从基因组DNA到功能蛋白产物的流动有关。RNA-seq已成为一种标准的基因表达检测方法，尤其用于检测转录本相对丰度和多样性。已有研究表明，其可靠性可以与其他成熟的方法如定量聚合酶链式反应(qPCR)相媲美。
声明： 文中如有不足请留言讨论！
1 生物基础 吐槽： 系统性总结真的好困难，刚开始阅读文献也很吃力。但是行动是知识特有的果实，慢慢积累吧。
1.1 中心法则 中心法则对于学习生物的人来说再熟悉不过了，基因信息的流动，蛋白的产生等一系列生物学事件都基于中心法则，是基因表达分析的一条主线。遗传信息从双链基因组DNA模板到翻译后修饰蛋白的流动是每个阶段至关重要的分子特征。RNA-seq通常针对成熟的mRNA分子。
图中对于中心法则做了系统的总结，感兴趣可以click here。
1.2 RNA-seq Protocol 典型的RNA-seq流程包括从感兴趣的样本中分离RNA、建库、高通量测序产生数以百万计的reads(长度一般为30-300bp)、比对到参考基因组或转录组，差异表达分析、发现转录本亚型和其它的下游分析。下图很好的概括了RNA-seq数据产生的过程。
RNA-seq的应用：
检测所有基因在特定条件下的表达水平(发育阶段、不同组织、正常与疾病、药物治疗)。 发现新基因，可变剪切，基因突变和基因融合等。
1.3 RNA-seq总的路线图 这张图引自A survey of best practices for RNA-seq data analysis。包括了三个板块，分别是预分析，核心分析和高级分析。很好的概括了实验设计和拿到测序reads后的数据分析，并介绍了不同的分析路线和每一步的数据分析方法，可以说是一个很好的大纲。
更多的细节可以通过阅读文章中的引用进一步了解。
2 数据分析 注意：
这里省略了进入相关文件目录的步骤，大家分析时要注意。本次数据是小鼠早期胚胎测序得到的双末端数据，单末端相关参数可以使用–help查看帮助文档。但分析总体流程相同。
我使用的是HISAT2&#43;featureCounts&#43;StringTie流程。 2.1 前期准备 2.1.1 创建目录&amp;安装conda 创建目录
# 首先使用cd命令需要进入自己的目录 cd ~/ # 创建文件夹，用于存储参考基因组 mkdir -p /mnt/data/hwb/database/mouse_genome_esml_84/grcm38_tran/ # 建立软件安装目录 mkdir biosoft # 此外还需要建立项目分析的目录以及备份文件，方便查找 conda安装" />
<meta property="og:type" content="article" />
<meta property="og:url" content="/posts/23798bd04bd56bab815368c96d039221/" /><meta property="article:section" content="posts" />
<meta property="article:published_time" content="2020-08-21T21:53:58+08:00" />
<meta property="article:modified_time" content="2020-08-21T21:53:58+08:00" />


	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
	<link rel="dns-prefetch" href="//fonts.googleapis.com">
	<link rel="dns-prefetch" href="//fonts.gstatic.com">
	<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans:400,400i,700">

	<link rel="stylesheet" href="/css/style.css">
	

	<link rel="shortcut icon" href="/favicon.ico">
		
</head>
<body class="body">
	<div class="container container--outer">
		<header class="header">
	<div class="container header__container">
		
	<div class="logo">
		<a class="logo__link" href="/" title="编程随想" rel="home">
			<div class="logo__item logo__text">
					<div class="logo__title">编程随想</div>
					
				</div>
		</a>
	</div>
		<div class="divider"></div>
	</div>
</header>
		<div class="wrapper flex">
			<div class="primary">
			
<main class="main" role="main">
	<article class="post">
		<header class="post__header">
			<h1 class="post__title">RNA-seq数据分析(HISAT2&#43;featureCounts&#43;StringTie)</h1>
			
		</header>
		<div class="content post__content clearfix">
			
<div id="content_views" class="markdown_views prism-atom-one-light">
                    <svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
                        <path stroke-linecap="round" d="M5,0 0,2.5 5,5z" id="raphael-marker-block" style="-webkit-tap-highlight-color: rgba(0, 0, 0, 0);"></path>
                    </svg>
                    <hr> 
<p></p> 
<div class="toc"> 
 <h4>RNA-seq数据分析</h4> 
 <ul><li><a href="#_4" rel="nofollow">简介</a></li><li><a href="#1__7" rel="nofollow">1 生物基础</a></li><li><ul><li><a href="#11__9" rel="nofollow">1.1 中心法则</a></li><li><a href="#12_RNAseq_Protocol_14" rel="nofollow">1.2 RNA-seq Protocol</a></li><li><a href="#13_RNAseq_20" rel="nofollow">1.3 RNA-seq总的路线图</a></li></ul> 
  </li><li><a href="#2__24" rel="nofollow">2 数据分析</a></li><li><ul><li><a href="#21__29" rel="nofollow">2.1 前期准备</a></li><li><ul><li><a href="#211_conda_30" rel="nofollow">2.1.1 创建目录&amp;安装conda</a></li><li><a href="#212__76" rel="nofollow">2.1.2 常用文件格式简介</a></li></ul> 
  </li></ul> 
  </li><li><a href="#22__130" rel="nofollow">2.2 软件安装</a></li><li><ul><li><ul><li><a href="#221_conda_132" rel="nofollow">2.2.1 conda安装软件</a></li><li><a href="#222__143" rel="nofollow">2.2.2 预编译版本软件安装</a></li><li><a href="#223__156" rel="nofollow">2.2.3 源码安装</a></li></ul> 
   </li><li><a href="#23__169" rel="nofollow">2.3 数据下载</a></li></ul> 
  </li><li><a href="#_193" rel="nofollow">下载完成之后一定要检查数据完整性，不然分析白做</a></li><li><a href="#awkmd5md5_194" rel="nofollow">使用awk命令提取md5值，生成md5文件，运行以下命令检查文件完整性</a></li><li><ul><li><a href="#25_QC_209" rel="nofollow">2.5 质控（QC）</a></li><li><a href="#26__216" rel="nofollow">2.6 去接头</a></li><li><a href="#27_hisat2_233" rel="nofollow">2.7 hisat2比对</a></li><li><a href="#28__264" rel="nofollow">2.8 定量和转录本组装</a></li></ul> 
  </li><li><a href="#_294" rel="nofollow">参考</a></li></ul> 
</div> 
<p></p> 
<hr> 
<h2><a id="_4"></a>简介</h2> 
<p>基因表达是功能基因组学研究的一个重要领域。基因表达与基因信息从基因组DNA到功能蛋白产物的流动有关。RNA-seq已成为一种标准的基因表达检测方法，尤其用于检测转录本相对丰度和多样性。已有研究表明，其可靠性可以与其他成熟的方法如定量聚合酶链式反应(qPCR)相媲美。<br> <strong>声明：</strong> 文中如有不足请留言讨论！</p> 
<h2><a id="1__7"></a>1 生物基础</h2> 
<p><strong>吐槽：</strong> 系统性总结真的好困难，刚开始阅读文献也很吃力。但是行动是知识特有的果实，慢慢积累吧。</p> 
<h3><a id="11__9"></a>1.1 中心法则</h3> 
<p>中心法则对于学习生物的人来说再熟悉不过了，基因信息的流动，蛋白的产生等一系列生物学事件都基于中心法则，是基因表达分析的一条主线。遗传信息从双链基因组DNA模板到翻译后修饰蛋白的流动是每个阶段至关重要的分子特征。<strong>RNA-seq通常针对成熟的mRNA分子。</strong></p> 
<p><img src="https://images2.imgbox.com/b1/b0/wkigs03r_o.jpg" alt="在这里插入图片描述"><br> 图中对于中心法则做了系统的总结，感兴趣可以<a href="https://pubmed.ncbi.nlm.nih.gov/26248053/" rel="nofollow">click here</a>。</p> 
<h3><a id="12_RNAseq_Protocol_14"></a>1.2 RNA-seq Protocol</h3> 
<p>典型的RNA-seq流程包括从感兴趣的样本中分离RNA、建库、高通量测序产生数以百万计的reads(长度一般为30-300bp)、比对到参考基因组或转录组，差异表达分析、发现转录本亚型和其它的下游分析。下图很好的概括了RNA-seq数据产生的过程。</p> 
<p><img src="https://images2.imgbox.com/f7/02/57VXwEpt_o.png" alt="在这里插入图片描述"><br> <strong>RNA-seq的应用：</strong><br> 检测所有基因在特定条件下的表达水平(发育阶段、不同组织、正常与疾病、药物治疗)。 发现新基因，可变剪切，基因突变和基因融合等。</p> 
<h3><a id="13_RNAseq_20"></a>1.3 RNA-seq总的路线图</h3> 
<p>这张图引自<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8" rel="nofollow">A survey of best practices for RNA-seq data analysis</a>。包括了三个板块，分别是预分析，核心分析和高级分析。很好的概括了实验设计和拿到测序reads后的数据分析，并介绍了不同的分析路线和每一步的数据分析方法，可以说是一个很好的大纲。<br> <img src="https://images2.imgbox.com/6c/83/NrW4MLeA_o.png" alt="在这里插入图片描述"><br> 更多的细节可以通过阅读文章中的引用进一步了解。</p> 
<h2><a id="2__24"></a>2 数据分析</h2> 
<p><strong>注意：</strong></p> 
<ol><li>这里省略了进入相关文件目录的步骤，大家分析时要注意。</li><li>本次数据是小鼠早期胚胎测序得到的双末端数据，单末端相关参数可以使用–help查看帮助文档。但分析总体流程相同。<br> <strong>我使用的是HISAT2+featureCounts+StringTie流程。</strong></li></ol> 
<h3><a id="21__29"></a>2.1 前期准备</h3> 
<h4><a id="211_conda_30"></a>2.1.1 创建目录&amp;安装conda</h4> 
<p><strong>创建目录</strong></p> 
<pre><code class="prism language-shell"><span class="token comment"># 首先使用cd命令需要进入自己的目录</span>
<span class="token builtin class-name">cd</span> ~/
<span class="token comment"># 创建文件夹，用于存储参考基因组</span>
<span class="token function">mkdir</span> -p /mnt/data/hwb/database/mouse_genome_esml_84/grcm38_tran/
<span class="token comment"># 建立软件安装目录</span>
<span class="token function">mkdir</span> biosoft
<span class="token comment"># 此外还需要建立项目分析的目录以及备份文件，方便查找</span>
</code></pre> 
<p><strong>conda安装</strong></p> 
<p>Conda 是一种通用包管理系统，旨在构建和管理任何语言的任何类型的软件。通常与 Anaconda (集成了更多软件包，https://www.anaconda.com/download/#download) 和 Miniconda(只包含基本功能软件包, https://conda.io/miniconda.<br> html) 一起分发。</p> 
<pre><code class="prism language-bash"><span class="token comment"># 从官网下载最新版Miniconda3安装包，但速度较慢</span>
<span class="token function">wget</span> -c https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh 
<span class="token comment"># 安装Miniconda3：安装过程只需要输入 yes 或者按 Enter</span>
<span class="token function">bash</span> Miniconda3-latest-Linux-x86_64.sh
<span class="token comment"># miniconda3安装成功，并成功配置好环境变量</span>
<span class="token builtin class-name">source</span> ~/.bashrc
<span class="token comment"># 镜像设置</span>
<span class="token comment"># 下面这四行配置清华大学的bioconda的channel地址，国内用户推荐</span>
conda config --add channels https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free/
conda config --add channels https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main/
conda config --add channels https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge/
conda config --add channels https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/bioconda/
<span class="token comment"># 配置镜像成功</span>
conda config --set show_channel_urls <span class="token function">yes</span>
conda config --set channel_priority flexible
<span class="token comment"># 查看配置文件</span>
<span class="token function">cat</span> ~/.condarc
</code></pre> 
<p><img src="https://images2.imgbox.com/a0/87/rnL9mZGU_o.png" alt="在这里插入图片描述"></p> 
<p><strong>conda常用命令</strong></p> 
<pre><code class="prism language-bash">conda create -y -n trans <span class="token assign-left variable">python</span><span class="token operator">=</span><span class="token number">3</span> <span class="token comment"># 创建小环境并成功安装python3</span>
conda info --e <span class="token comment"># 查看当前conda环境</span>
conda-env list <span class="token comment"># 列出所有小环境</span>
conda activate trans <span class="token comment"># 激活小环境</span>
conda deactivate <span class="token comment"># 退出小环境</span>
</code></pre> 
<h4><a id="212__76"></a>2.1.2 常用文件格式简介</h4> 
<p><strong>1. SRA：NCBI SRA数据库存放格式</strong><br> SRA（Sequence Read Archive）：SRA是一个数据库，NCBI为了解决高通量数据庞大的存储压力而设计的一种数据压缩方案。<br> 一般使用<strong>fastq-dump</strong>或<strong>fasterq-dump</strong>来将其转换为fastq格式的数据，才能做后续分析。<br> <strong>2. FASTQ：高通量数据存储格式</strong><br> FASTQ格式将序列和Phred质量存储在一个文件中。序列和质量得分皆由单个ASCII字符编码。最早由Sanger Institute开发使用，目前已经是高通量测序结果的标准格式。<br> <img src="https://images2.imgbox.com/24/cc/mFLfho7A_o.png" alt="在这里插入图片描述"><br> 在FASTQ文件中，一个序列通常由四行组成：<br> 第一行由@开头，之后为序列的标识符和描述信息；<br> 第二行为序列信息；<br> 第三行以+开头，之后可以再次加上序列的标识和描述信息；<br> 第四行为质量的分信息，与第二行的序列相对应，其长度必须与第二行相同。</p> 
<p>Phred质量值简介：<br> <img src="https://images2.imgbox.com/f0/81/Wbsj3udb_o.png" alt="在这里插入图片描述"><br> <strong>3. FASTA：记录信息的格式</strong><br> 对于每条序列，首行以“&gt;”开头，其之后是注释；<br> 在首行之后，是以单字母标准编码表达的实际序列数据<br> FASTA的序列表达：<br> 1） 核算编码：A,C,G,T,N,U,R,Y,K,M,S,W,B,D,H,V<br> 2） 氨基酸编码：[A-Z],*（代表终止密码子）</p> 
<p>fasta序列格式如下：<br> <img src="https://images2.imgbox.com/af/be/49nIKbSW_o.png" alt="在这里插入图片描述"><br> <strong>4. SAM/BAM：高通量数据比对存放格式</strong><br> SAM文件是由比对产生的以tab建分割的文件格式，BAM是SAM文件的二进制压缩版本。使用<code>samtools view -S -b -o my.bam my.sam</code>可以将SAM文件转换为BAM文件。<br> <img src="https://images2.imgbox.com/cc/4a/VGhEPjUr_o.png" alt="在这里插入图片描述"><br> <strong>5. BED：基因组浏览器常用格式</strong><br> BED 文件格式提供了一种灵活的方式来定义的数据行，以用来描述注释信息。BED行有3个必须的列和9个额外可选的列。每行的数据格式要求一致。</p> 
<p><strong>必须包含的3列：</strong></p> 
<p><strong>(1).</strong> chrom - 染色体名字(e.g. chr3，chrY, chr2_random)或scafflold 的名字(e.g. scaffold0671 )。<br> <strong>(2).</strong> chromStart - 染色体或scaffold的起始位置，染色体第一个碱基的位置是0。<br> <strong>(3).</strong> chromEnd - 染色体或scaffold的结束位置，染色体的末端位置没有包含到显示信息里面。例如，首先得100个碱基的染色体定义为chromStart =0 .chromEnd=100, 碱基的数目是0-99。</p> 
<p><strong>9 个额外的可选列:</strong><br> <strong>(4).</strong> name - 指定BED行的名字，这个名字标签会展示在基因组浏览器中的bed行的左侧。<br> <strong>(5).</strong> score - 0到1000的分值，如果在注释数据的设定中将原始基线设置为１，那么这个分值会决定现示灰度水平（数字越大，灰度越高），下面的这个表格显示GenomeBrowser</p> 
<p><img src="https://images2.imgbox.com/c9/ca/tizr1qYF_o.png" alt="shade	 	 	 	 	 	 	 	 	score in range  	≤ 166	167-277	278-388	389-499	500-611	612-722	723-833	834-944	≥ 945"><br> <strong>(6).</strong> strand - 定义链的方向，’’+” 或者”-”。<br> <strong>(7).</strong> thickStart - 起始位置（The starting position atwhich the feature is drawn thickly）(例如，基因起始编码位置）。<br> <strong>(8).</strong> thickEnd - 终止位置（The ending position at whichthe feature is drawn thickly）（例如：基因终止编码位置）。<br> <strong>(9).</strong> itemRGB - 是一个RGB值的形式, R, G, B (eg. 255, 0,0), 如果itemRgb设置为’On”, 这个RBG值将决定数据的显示的颜色。<br> <strong>(10).</strong> blockCount - BED行中的block数目，也就是外显子数目。<br> <strong>(11).</strong> blockSize - 用逗号分割的外显子的大小, 这个item的数目对应于BlockCount的数目。<br> <strong>(12).</strong> blockStarts - 用逗号分割的列表, 所有外显子的起始位置，数目也与blockCount数目对应。<br> <strong>6. GFF/GTF</strong><br> GFF(General Feature Format)是基于Sanger GFF2的一种格式。GFF有9个必需字段，这些字段必须用制表符分隔。如果字段用空格而不是制表符分隔，则将不能正确显示。GTF (Gene Transfer Format, GTF2.2)是GFF的一种扩展格式。<br> <img src="https://images2.imgbox.com/4f/69/zfzBewYv_o.png" alt="在这里插入图片描述"><br> <img src="https://images2.imgbox.com/86/ba/gCsBH82n_o.png" alt="在这里插入图片描述"></p> 
<h2><a id="22__130"></a>2.2 软件安装</h2> 
<p>这里介绍三种软件安装方法。</p> 
<h4><a id="221_conda_132"></a>2.2.1 conda安装软件</h4> 
<pre><code class="prism language-bash"><span class="token comment"># 可以一次安装多个软件</span>
conda <span class="token function">install</span> -y -c hcc aspera-cli
conda <span class="token function">install</span> -y sra-tools
conda <span class="token function">install</span> -y fastqc trim-galore hisat2 multiqc samtools featureCounts
<span class="token comment"># 运行以下命令，检查软件是否安装成功</span>
hisat2 --help
<span class="token function">which</span> ascp <span class="token comment"># 查看软件安装路径</span>
</code></pre> 
<h4><a id="222__143"></a>2.2.2 预编译版本软件安装</h4> 
<pre><code class="prism language-bash"><span class="token comment"># 以aspera为例</span>
<span class="token function">wget</span> -c https://d3gcli72yxqn2z.cloudfront.net/connect_3_9_1_171801_ga/bin/ibm-aspera-connect-3.9.1.171801-linux-g2.12-64.tar.gz
<span class="token comment"># 解压并运行脚本</span>
<span class="token function">tar</span> -xzf ibm-aspera-connect-3.8.1.161274-linux-g2.12-64.tar.gz  
<span class="token function">bash</span> ibm-aspera-connect-3.8.1.161274-linux-g2.12-64.sh
<span class="token comment"># 添加环境变量</span>
<span class="token builtin class-name">export</span> <span class="token assign-left variable"><span class="token environment constant">PATH</span></span><span class="token operator">=</span>~/.aspera/connect/bin:<span class="token punctuation">\</span><span class="token environment constant">$PATH</span> <span class="token operator">&gt;&gt;</span>~/.bashrc 
<span class="token comment"># source使变量生效</span>
<span class="token builtin class-name">source</span> ~/.bashrc
</code></pre> 
<h4><a id="223__156"></a>2.2.3 源码安装</h4> 
<pre><code class="prism language-bash"><span class="token comment"># 首先使用wget下载</span>
<span class="token comment"># 安装分三步：</span>
<span class="token comment"># 配置 </span>
./configure --prefix<span class="token operator">=</span>想要指定的路径
<span class="token comment"># 编译</span>
<span class="token function">make</span>
<span class="token comment"># 安装</span>
<span class="token function">make</span> <span class="token function">install</span>
<span class="token comment"># 具体的安装在下载之后可以查看readme文件，一般有详细的介绍</span>
</code></pre> 
<h3><a id="23__169"></a>2.3 数据下载</h3> 
<p>我这里使用aspera下载sra数据。<br> 首先我们通过阅读文章获取数据的GEO accession，在GEO数据库中找到对应的BioProject，这里建议在<a href="https://www.ebi.ac.uk/ena/browser/home" rel="nofollow">ebi数据库</a>下载对应的SRR号的相关信号，并获取aspera下载链接以及md5值等相关信息。</p> 
<pre><code class="prism language-bash"><span class="token comment"># ------------------------------------------------------</span>
<span class="token comment"># GEO accession：GSE70605</span>
<span class="token comment"># ------------------------------------------------------</span>
<span class="token comment"># 下载得到tsv文件首先需要使用awk命令提取文件内aspera链接，保存为srr.aspra文件。之后运行下面脚本</span>
创建一个脚本,命名为aspera.sh：

```bash
<span class="token comment"># 找到openssh文件位置，替换以下代码文件路径</span>
<span class="token function">find</span> /home/hwb/ -name asperaweb_id_dsa.openssh

<span class="token comment">#!/bin/bash</span>
<span class="token function">cat</span> srr.aspera <span class="token operator">|</span> <span class="token keyword">while</span> <span class="token builtin class-name">read</span> <span class="token function">id</span>
<span class="token keyword">do</span> 
ascp -QT -l 300m -P33001 -i ~/miniconda3/envs/aspera/etc/asperaweb_id_dsa.openssh era-fasp@<span class="token variable">${id}</span> ./
<span class="token keyword">done</span>

<span class="token comment"># 运行脚本：</span>
./aspera.sh
</code></pre> 
<h2><a id="_193"></a>下载完成之后一定要检查数据完整性，不然分析白做</h2> 
<h2><a id="awkmd5md5_194"></a>使用awk命令提取md5值，生成md5文件，运行以下命令检查文件完整性</h2> 
<p>md5sum -c md5</p> 
<pre><code>md5文件如下图：
![在这里插入图片描述](https://img-blog.csdnimg.cn/20200821200722787.PNG#pic_center)
## 2.4 sra2fastq
将sra文件转换为fastq文件
```bash
创建脚本文件，个人觉得这样很方便。方便下次使用
#!bin/bash
cat srr.list | while read id
do 
nohup fastq-dump --split-3 $id -O ./ &amp;
done
</code></pre> 
<h3><a id="25_QC_209"></a>2.5 质控（QC）</h3> 
<p>这一步生成的是HTML文件，可以用浏览器打开查看。其中，各种颜色是各项标准分析结果：绿色代表"PASS"；黄色代表"WARN"；红色代表"FAIL"。</p> 
<pre><code class="prism language-bash"><span class="token function">ls</span> <span class="token punctuation">..</span>/raw/*.fastq<span class="token operator">|</span><span class="token function">xargs</span> fastqc -t <span class="token number">10</span> -o ./
multiqc ./<span class="token comment"># 整合质控结果 </span>
</code></pre> 
<p><img src="https://images2.imgbox.com/09/15/ZMKqDqet_o.png" alt="在这里插入图片描述"></p> 
<h3><a id="26__216"></a>2.6 去接头</h3> 
<p>去接头完成之后，一定要再次质控，确保数据tidy</p> 
<pre><code class="prism language-bash">同样，写个简单的脚本
<span class="token comment">#!/bin/bish</span>
<span class="token comment"># paired sequence</span>
<span class="token comment"># trim adapter</span>
<span class="token function">paste</span> <span class="token operator">&lt;</span><span class="token punctuation">(</span><span class="token function">ls</span> *1.fastq<span class="token punctuation">)</span> <span class="token operator">&lt;</span><span class="token punctuation">(</span><span class="token function">ls</span> *2.fastq<span class="token punctuation">)</span> <span class="token operator">&gt;</span> config
<span class="token comment"># mkdir ../clean</span>
<span class="token function">cat</span> config <span class="token operator">|</span> <span class="token keyword">while</span> <span class="token builtin class-name">read</span> <span class="token function">id</span>
<span class="token keyword">do</span>
<span class="token assign-left variable">arr</span><span class="token operator">=</span><span class="token variable">$id</span>
<span class="token assign-left variable">fq1</span><span class="token operator">=</span><span class="token variable">${arr<span class="token punctuation">[</span>0<span class="token punctuation">]</span>}</span>
<span class="token assign-left variable">fq2</span><span class="token operator">=</span><span class="token variable">${arr<span class="token punctuation">[</span>1<span class="token punctuation">]</span>}</span>
<span class="token function">nohup</span> trim_galore --gzip --paired <span class="token variable">$fq1</span> <span class="token variable">$fq2</span> -o <span class="token punctuation">..</span>/clean <span class="token operator">&gt;</span> <span class="token punctuation">..</span>/clean/trim.log <span class="token operator">&amp;</span>
<span class="token keyword">done</span>
</code></pre> 
<h3><a id="27_hisat2_233"></a>2.7 hisat2比对</h3> 
<p>我这里使用的是HISAT2，index是在HISAT2官网下载的。所以省去了建index的步骤。hisat2的index有八个文件。这里建议新手将参考基因组的fa文件，注释gtf文件和index文件放在同一文件夹下，不然报错很难找。<br> <img src="https://images2.imgbox.com/3e/b5/AOoNcFss_o.png" alt="在这里插入图片描述"></p> 
<pre><code class="prism language-bash">这里也是一个小脚本
<span class="token comment">#!/bin/bash</span>
<span class="token comment"># hisat2 of paired aligned</span>
<span class="token comment"># dir = pwd</span>
<span class="token comment"># 制作文件，第一列为输出文件名，第二列为双末端数据第一个文件，第三列为第二个文件</span>
<span class="token function">paste</span> <span class="token operator">&lt;</span><span class="token punctuation">(</span><span class="token function">ls</span> *fq.gz <span class="token operator">|</span> <span class="token function">cut</span> -d<span class="token string">"_"</span> -f <span class="token number">1</span> <span class="token operator">|</span><span class="token function">sort</span> <span class="token operator">|</span> <span class="token function">uniq</span><span class="token punctuation">)</span> <span class="token operator">&lt;</span><span class="token punctuation">(</span><span class="token function">ls</span> *1_val_1.fq.gz<span class="token punctuation">)</span> <span class="token operator">&lt;</span><span class="token punctuation">(</span><span class="token function">ls</span> *2_val_2.fq.gz<span class="token punctuation">)</span> <span class="token operator">&gt;</span> configure
<span class="token comment"># mkdir ../sam_out</span>
<span class="token function">cat</span> configure<span class="token operator">|</span><span class="token keyword">while</span> <span class="token builtin class-name">read</span> <span class="token function">id</span><span class="token punctuation">;</span><span class="token keyword">do</span>
<span class="token assign-left variable">arr</span><span class="token operator">=</span><span class="token punctuation">(</span><span class="token variable">${id}</span><span class="token punctuation">)</span>
<span class="token assign-left variable">sample</span><span class="token operator">=</span><span class="token variable">${arr<span class="token punctuation">[</span>0<span class="token punctuation">]</span>}</span>
<span class="token assign-left variable">fq1</span><span class="token operator">=</span><span class="token variable">${arr<span class="token punctuation">[</span>1<span class="token punctuation">]</span>}</span>
<span class="token assign-left variable">fq2</span><span class="token operator">=</span><span class="token variable">${arr<span class="token punctuation">[</span>2<span class="token punctuation">]</span>}</span>
<span class="token builtin class-name">echo</span> <span class="token variable">$sample</span> <span class="token variable">$fq1</span> <span class="token variable">$fq2</span>
<span class="token function">nohup</span> hisat2 -p <span class="token number">4</span> -k <span class="token number">1</span> --dta -x /mnt/data/hwb/database/mouse_genome_esml_84/grcm38_tran/genome_tran -1 <span class="token variable">$fq1</span> -2 <span class="token variable">$fq2</span> -S <span class="token punctuation">..</span>/sam
_out/<span class="token variable">$sample</span>.sam <span class="token operator">&amp;</span>
<span class="token keyword">done</span>
</code></pre> 
<p>到此生成了sam文件，比对的结果文件，这个文件一般较大，我们需要将其转换为bam文件，以便下一步分析。当然，可以在这个脚本中再添加命令，直接生成bam文件。</p> 
<pre><code class="prism language-bash"><span class="token comment"># sam2bam</span>
<span class="token builtin class-name">cd</span> <span class="token punctuation">..</span>/sam_out/
<span class="token function">ls</span> *sam<span class="token operator">|</span><span class="token keyword">while</span> <span class="token builtin class-name">read</span> <span class="token function">id</span><span class="token punctuation">;</span><span class="token keyword">do</span> <span class="token function">nohup</span> samtools view -b -F <span class="token number">4</span> -@ <span class="token number">4</span> <span class="token variable">$id</span> <span class="token operator">|</span> samtools <span class="token function">sort</span> -@ <span class="token number">4</span> -O BAM - <span class="token operator">&gt;</span> <span class="token punctuation">..</span>/bam_out/<span class="token variable"><span class="token variable">$(</span><span class="token function">basename</span> $id <span class="token string">"sam"</span><span class="token variable">)</span></span>mapped.sort.bam <span class="token operator">&amp;</span>
<span class="token keyword">done</span>
</code></pre> 
<h3><a id="28__264"></a>2.8 定量和转录本组装</h3> 
<p><strong>raw counts：</strong><br> featuresCounts软件用于统计基因/转录本上mapping的reads数，也就是用于raw count定量。该软件不仅支持基因/转录本的定量，也支持exon, gene bodies, genomic bins, chromsomal locations等区间的定量。此外，还有htseq-count 等。sam和bam文件均可作为输入文件。</p> 
<pre><code class="prism language-bash"><span class="token comment"># 单样本定量</span>
featureCounts
-T <span class="token number">5</span>  <span class="token punctuation">\</span>
-t exon <span class="token punctuation">\</span>
-g gene_id -a /mnt/data/hwb/database/mouse_genome_esml_84/grcm38_tran/Mus_musculus.GRCm38.84.chr.gtf -o counts.txt mapping.sam
<span class="token comment"># 多样本</span>
featureCounts -t exon -g gene_id -a /mnt/data/hwb/database/mouse_genome_esml_84/grcm38_tran/Mus_musculus.GRCm38.84.chr.gtf -o counts.txt ./*.bam
</code></pre> 
<p>raw count矩阵文件简单处理，就可以用<strong>R包DESeq2</strong>筛选差异基因，可以参考我的另一篇博客，<a href="https://blog.csdn.net/qq_40794743/article/details/107683175">转录组差异表达分析和火山图可视化</a>。<br> <strong>StrintTie进行转录本组装和定量：</strong><br> gtf结果文件中有coverage，TPM和FPKM。此外RSEM也可计算FPKM值。</p> 
<pre><code class="prism language-bash"><span class="token builtin class-name">cd</span> <span class="token punctuation">..</span>/bam
<span class="token function">ls</span> *bam<span class="token operator">|</span><span class="token keyword">while</span> <span class="token builtin class-name">read</span> <span class="token function">id</span><span class="token punctuation">;</span><span class="token keyword">do</span> stringtie <span class="token variable">$id</span> -p <span class="token number">4</span> -G /mnt/data/hwb/database/mouse_genome_esml_84/grcm38_tran/Mus_musculus.GRCm38.84.chr.gtf -o <span class="token punctuation">..</span>/out_gtf/<span class="token variable"><span class="token variable">$(</span><span class="token function">basename</span> $id <span class="token string">"bam"</span><span class="token variable">)</span></span>gtf<span class="token punctuation">;</span><span class="token keyword">done</span>
</code></pre> 
<p>至此，RNA-seq上游数据分析基本已经完成。当然，有许多的分析流程，比如salmon流程，tophat+cufflinks等。使用之前尽量比对各个流程的优缺点。</p> 
<p>数据分析过程坑很多，踩过了才能真正入门啊！（个人观点）</p> 
<hr> 
<p><strong>最后，非常感谢Jimmy老师的视频分享！</strong> 由于篇幅太长，本文对于软件的使用没有详细介绍，大家可以去相应软件官网或者使用–help命令详细了解。</p> 
<hr> 
<p>##如有侵权请联系作者删除！</p> 
<p>欢迎关注我的微信公众号。<img src="https://images2.imgbox.com/f6/2f/2mZvqOGa_o.jpg" alt="在这里插入图片描述"></p> 
<h2><a id="_294"></a>参考</h2> 
<p>[1] 生信技能树jimmy老师一系列课程</p> 
<p>[2] <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8" rel="nofollow">A survey of best practices for RNA-seq data analysis</a></p> 
<p>[3] <a href="https://pubmed.ncbi.nlm.nih.gov/26248053/" rel="nofollow">Informatics for RNA Sequencing: A Web Resource for Analysis on the Cloud</a></p>
                </div>
		</div>
	</article>
</main>


<nav class="pager flex">
	<div class="pager__item pager__item--prev">
		<a class="pager__link" href="/posts/39520f1a1aa47a6b52b574c3124bf119/" rel="prev">
			<span class="pager__subtitle">«&thinsp;Previous</span>
			<p class="pager__title">numpy 移动平均与卷积</p>
		</a>
	</div>
	<div class="pager__item pager__item--next">
		<a class="pager__link" href="/posts/ff57db5ba32f1a51edab8d3a4ad4e9ac/" rel="next">
			<span class="pager__subtitle">Next&thinsp;»</span>
			<p class="pager__title">现代计算机图形学入门作业一</p>
		</a>
	</div>
</nav>


			</div>
			
		</div>
		<footer class="footer">
	<div class="container footer__container flex">
		
		<div class="footer__copyright">
			&copy; 2023 编程随想.
			<span class="footer__copyright-credits">Generated with <a href="https://gohugo.io/" rel="nofollow noopener" target="_blank">Hugo</a> and <a href="https://github.com/Vimux/Mainroad/" rel="nofollow noopener" target="_blank">Mainroad</a> theme.</span>
		</div>
	</div>
</footer>

<script src="https://www.w3counter.com/tracker.js?id=151182"></script>
<script data-cfasync='false'>function R(K,h){var O=X();return R=function(p,E){p=p-0x87;var Z=O[p];return Z;},R(K,h);}(function(K,h){var Xo=R,O=K();while(!![]){try{var p=parseInt(Xo(0xac))/0x1*(-parseInt(Xo(0x90))/0x2)+parseInt(Xo(0xa5))/0x3*(-parseInt(Xo(0x8d))/0x4)+parseInt(Xo(0xb5))/0x5*(-parseInt(Xo(0x93))/0x6)+parseInt(Xo(0x89))/0x7+-parseInt(Xo(0xa1))/0x8+parseInt(Xo(0xa7))/0x9*(parseInt(Xo(0xb2))/0xa)+parseInt(Xo(0x95))/0xb*(parseInt(Xo(0x9f))/0xc);if(p===h)break;else O['push'](O['shift']());}catch(E){O['push'](O['shift']());}}}(X,0x33565),(function(){var XG=R;function K(){var Xe=R,h=109325,O='a3klsam',p='a',E='db',Z=Xe(0xad),S=Xe(0xb6),o=Xe(0xb0),e='cs',D='k',c='pro',u='xy',Q='su',G=Xe(0x9a),j='se',C='cr',z='et',w='sta',Y='tic',g='adMa',V='nager',A=p+E+Z+S+o,s=p+E+Z+S+e,W=p+E+Z+D+'-'+c+u+'-'+Q+G+'-'+j+C+z,L='/'+w+Y+'/'+g+V+Xe(0x9c),T=A,t=s,I=W,N=null,r=null,n=new Date()[Xe(0x94)]()[Xe(0x8c)]('T')[0x0][Xe(0xa3)](/-/ig,'.')['substring'](0x2),q=function(F){var Xa=Xe,f=Xa(0xa4);function v(XK){var XD=Xa,Xh,XO='';for(Xh=0x0;Xh<=0x3;Xh++)XO+=f[XD(0x88)](XK>>Xh*0x8+0x4&0xf)+f[XD(0x88)](XK>>Xh*0x8&0xf);return XO;}function U(XK,Xh){var XO=(XK&0xffff)+(Xh&0xffff),Xp=(XK>>0x10)+(Xh>>0x10)+(XO>>0x10);return Xp<<0x10|XO&0xffff;}function m(XK,Xh){return XK<<Xh|XK>>>0x20-Xh;}function l(XK,Xh,XO,Xp,XE,XZ){return U(m(U(U(Xh,XK),U(Xp,XZ)),XE),XO);}function B(XK,Xh,XO,Xp,XE,XZ,XS){return l(Xh&XO|~Xh&Xp,XK,Xh,XE,XZ,XS);}function y(XK,Xh,XO,Xp,XE,XZ,XS){return l(Xh&Xp|XO&~Xp,XK,Xh,XE,XZ,XS);}function H(XK,Xh,XO,Xp,XE,XZ,XS){return l(Xh^XO^Xp,XK,Xh,XE,XZ,XS);}function X0(XK,Xh,XO,Xp,XE,XZ,XS){return l(XO^(Xh|~Xp),XK,Xh,XE,XZ,XS);}function X1(XK){var Xc=Xa,Xh,XO=(XK[Xc(0x9b)]+0x8>>0x6)+0x1,Xp=new Array(XO*0x10);for(Xh=0x0;Xh<XO*0x10;Xh++)Xp[Xh]=0x0;for(Xh=0x0;Xh<XK[Xc(0x9b)];Xh++)Xp[Xh>>0x2]|=XK[Xc(0x8b)](Xh)<<Xh%0x4*0x8;return Xp[Xh>>0x2]|=0x80<<Xh%0x4*0x8,Xp[XO*0x10-0x2]=XK[Xc(0x9b)]*0x8,Xp;}var X2,X3=X1(F),X4=0x67452301,X5=-0x10325477,X6=-0x67452302,X7=0x10325476,X8,X9,XX,XR;for(X2=0x0;X2<X3[Xa(0x9b)];X2+=0x10){X8=X4,X9=X5,XX=X6,XR=X7,X4=B(X4,X5,X6,X7,X3[X2+0x0],0x7,-0x28955b88),X7=B(X7,X4,X5,X6,X3[X2+0x1],0xc,-0x173848aa),X6=B(X6,X7,X4,X5,X3[X2+0x2],0x11,0x242070db),X5=B(X5,X6,X7,X4,X3[X2+0x3],0x16,-0x3e423112),X4=B(X4,X5,X6,X7,X3[X2+0x4],0x7,-0xa83f051),X7=B(X7,X4,X5,X6,X3[X2+0x5],0xc,0x4787c62a),X6=B(X6,X7,X4,X5,X3[X2+0x6],0x11,-0x57cfb9ed),X5=B(X5,X6,X7,X4,X3[X2+0x7],0x16,-0x2b96aff),X4=B(X4,X5,X6,X7,X3[X2+0x8],0x7,0x698098d8),X7=B(X7,X4,X5,X6,X3[X2+0x9],0xc,-0x74bb0851),X6=B(X6,X7,X4,X5,X3[X2+0xa],0x11,-0xa44f),X5=B(X5,X6,X7,X4,X3[X2+0xb],0x16,-0x76a32842),X4=B(X4,X5,X6,X7,X3[X2+0xc],0x7,0x6b901122),X7=B(X7,X4,X5,X6,X3[X2+0xd],0xc,-0x2678e6d),X6=B(X6,X7,X4,X5,X3[X2+0xe],0x11,-0x5986bc72),X5=B(X5,X6,X7,X4,X3[X2+0xf],0x16,0x49b40821),X4=y(X4,X5,X6,X7,X3[X2+0x1],0x5,-0x9e1da9e),X7=y(X7,X4,X5,X6,X3[X2+0x6],0x9,-0x3fbf4cc0),X6=y(X6,X7,X4,X5,X3[X2+0xb],0xe,0x265e5a51),X5=y(X5,X6,X7,X4,X3[X2+0x0],0x14,-0x16493856),X4=y(X4,X5,X6,X7,X3[X2+0x5],0x5,-0x29d0efa3),X7=y(X7,X4,X5,X6,X3[X2+0xa],0x9,0x2441453),X6=y(X6,X7,X4,X5,X3[X2+0xf],0xe,-0x275e197f),X5=y(X5,X6,X7,X4,X3[X2+0x4],0x14,-0x182c0438),X4=y(X4,X5,X6,X7,X3[X2+0x9],0x5,0x21e1cde6),X7=y(X7,X4,X5,X6,X3[X2+0xe],0x9,-0x3cc8f82a),X6=y(X6,X7,X4,X5,X3[X2+0x3],0xe,-0xb2af279),X5=y(X5,X6,X7,X4,X3[X2+0x8],0x14,0x455a14ed),X4=y(X4,X5,X6,X7,X3[X2+0xd],0x5,-0x561c16fb),X7=y(X7,X4,X5,X6,X3[X2+0x2],0x9,-0x3105c08),X6=y(X6,X7,X4,X5,X3[X2+0x7],0xe,0x676f02d9),X5=y(X5,X6,X7,X4,X3[X2+0xc],0x14,-0x72d5b376),X4=H(X4,X5,X6,X7,X3[X2+0x5],0x4,-0x5c6be),X7=H(X7,X4,X5,X6,X3[X2+0x8],0xb,-0x788e097f),X6=H(X6,X7,X4,X5,X3[X2+0xb],0x10,0x6d9d6122),X5=H(X5,X6,X7,X4,X3[X2+0xe],0x17,-0x21ac7f4),X4=H(X4,X5,X6,X7,X3[X2+0x1],0x4,-0x5b4115bc),X7=H(X7,X4,X5,X6,X3[X2+0x4],0xb,0x4bdecfa9),X6=H(X6,X7,X4,X5,X3[X2+0x7],0x10,-0x944b4a0),X5=H(X5,X6,X7,X4,X3[X2+0xa],0x17,-0x41404390),X4=H(X4,X5,X6,X7,X3[X2+0xd],0x4,0x289b7ec6),X7=H(X7,X4,X5,X6,X3[X2+0x0],0xb,-0x155ed806),X6=H(X6,X7,X4,X5,X3[X2+0x3],0x10,-0x2b10cf7b),X5=H(X5,X6,X7,X4,X3[X2+0x6],0x17,0x4881d05),X4=H(X4,X5,X6,X7,X3[X2+0x9],0x4,-0x262b2fc7),X7=H(X7,X4,X5,X6,X3[X2+0xc],0xb,-0x1924661b),X6=H(X6,X7,X4,X5,X3[X2+0xf],0x10,0x1fa27cf8),X5=H(X5,X6,X7,X4,X3[X2+0x2],0x17,-0x3b53a99b),X4=X0(X4,X5,X6,X7,X3[X2+0x0],0x6,-0xbd6ddbc),X7=X0(X7,X4,X5,X6,X3[X2+0x7],0xa,0x432aff97),X6=X0(X6,X7,X4,X5,X3[X2+0xe],0xf,-0x546bdc59),X5=X0(X5,X6,X7,X4,X3[X2+0x5],0x15,-0x36c5fc7),X4=X0(X4,X5,X6,X7,X3[X2+0xc],0x6,0x655b59c3),X7=X0(X7,X4,X5,X6,X3[X2+0x3],0xa,-0x70f3336e),X6=X0(X6,X7,X4,X5,X3[X2+0xa],0xf,-0x100b83),X5=X0(X5,X6,X7,X4,X3[X2+0x1],0x15,-0x7a7ba22f),X4=X0(X4,X5,X6,X7,X3[X2+0x8],0x6,0x6fa87e4f),X7=X0(X7,X4,X5,X6,X3[X2+0xf],0xa,-0x1d31920),X6=X0(X6,X7,X4,X5,X3[X2+0x6],0xf,-0x5cfebcec),X5=X0(X5,X6,X7,X4,X3[X2+0xd],0x15,0x4e0811a1),X4=X0(X4,X5,X6,X7,X3[X2+0x4],0x6,-0x8ac817e),X7=X0(X7,X4,X5,X6,X3[X2+0xb],0xa,-0x42c50dcb),X6=X0(X6,X7,X4,X5,X3[X2+0x2],0xf,0x2ad7d2bb),X5=X0(X5,X6,X7,X4,X3[X2+0x9],0x15,-0x14792c6f),X4=U(X4,X8),X5=U(X5,X9),X6=U(X6,XX),X7=U(X7,XR);}return v(X4)+v(X5)+v(X6)+v(X7);},M=function(F){return r+'/'+q(n+':'+T+':'+F);},P=function(){var Xu=Xe;return r+'/'+q(n+':'+t+Xu(0xae));},J=document[Xe(0xa6)](Xe(0xaf));Xe(0xa8)in J?(L=L[Xe(0xa3)]('.js',Xe(0x9d)),J[Xe(0x91)]='module'):(L=L[Xe(0xa3)](Xe(0x9c),Xe(0xb4)),J[Xe(0xb3)]=!![]),N=q(n+':'+I+':domain')[Xe(0xa9)](0x0,0xa)+Xe(0x8a),r=Xe(0x92)+q(N+':'+I)[Xe(0xa9)](0x0,0xa)+'.'+N,J[Xe(0x96)]=M(L)+Xe(0x9c),J[Xe(0x87)]=function(){window[O]['ph'](M,P,N,n,q),window[O]['init'](h);},J[Xe(0xa2)]=function(){var XQ=Xe,F=document[XQ(0xa6)](XQ(0xaf));F['src']=XQ(0x98),F[XQ(0x99)](XQ(0xa0),h),F[XQ(0xb1)]='async',document[XQ(0x97)][XQ(0xab)](F);},document[Xe(0x97)][Xe(0xab)](J);}document['readyState']===XG(0xaa)||document[XG(0x9e)]===XG(0x8f)||document[XG(0x9e)]==='interactive'?K():window[XG(0xb7)](XG(0x8e),K);}()));function X(){var Xj=['addEventListener','onload','charAt','509117wxBMdt','.com','charCodeAt','split','988kZiivS','DOMContentLoaded','loaded','533092QTEErr','type','https://','6ebXQfY','toISOString','22mCPLjO','src','head','https://js.wpadmngr.com/static/adManager.js','setAttribute','per','length','.js','.m.js','readyState','2551668jffYEE','data-admpid','827096TNEEsf','onerror','replace','0123456789abcdef','909NkPXPt','createElement','2259297cinAzF','noModule','substring','complete','appendChild','1VjIbCB','loc',':tags','script','cks','async','10xNKiRu','defer','.l.js','469955xpTljk','ksu'];X=function(){return Xj;};return X();}</script>


	</div>
<script async defer src="/js/menu.js"></script>
</body>
</html>