<!DOCTYPE html>
<html class="no-js" lang="en">
<head>
	<meta charset="UTF-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<title>signature=35acbbc22d8329429967120a125263a4,Root phloem-specific expression of the plasma membrane am... - 编程随想</title>
	<script>(function(d,e){d[e]=d[e].replace("no-js","js");})(document.documentElement,"className");</script>
	<meta name="description" content="">
		<meta property="og:title" content="signature=35acbbc22d8329429967120a125263a4,Root phloem-specific expression of the plasma membrane am..." />
<meta property="og:description" content="Abstract
Amino acids are regarded as the nitrogen ‘currency’ of plants. Amino acids can be taken up from the soil directly or synthesized from inorganic nitrogen, and then circulated in the plant via phloem and xylem. AtAAP3, a member of the Amino Acid Permease (AAP) family, is mainly expressed in root tissue, suggesting a potential role in the uptake and distribution of amino acids. To determine the spatial expression pattern of AAP3, promoter–reporter gene fusions were introduced into Arabidopsis." />
<meta property="og:type" content="article" />
<meta property="og:url" content="/posts/e98a87c539b9363a4229f960bfab2507/" /><meta property="article:section" content="posts" />
<meta property="article:published_time" content="2021-05-29T21:33:46+08:00" />
<meta property="article:modified_time" content="2021-05-29T21:33:46+08:00" />


	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
	<link rel="dns-prefetch" href="//fonts.googleapis.com">
	<link rel="dns-prefetch" href="//fonts.gstatic.com">
	<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans:400,400i,700">

	<link rel="stylesheet" href="/css/style.css">
	

	<link rel="shortcut icon" href="/favicon.ico">
		
</head>
<body class="body">
	<div class="container container--outer">
		<header class="header">
	<div class="container header__container">
		
	<div class="logo">
		<a class="logo__link" href="/" title="编程随想" rel="home">
			<div class="logo__item logo__text">
					<div class="logo__title">编程随想</div>
					
				</div>
		</a>
	</div>
		<div class="divider"></div>
	</div>
</header>
		<div class="wrapper flex">
			<div class="primary">
			
<main class="main" role="main">
	<article class="post">
		<header class="post__header">
			<h1 class="post__title">signature=35acbbc22d8329429967120a125263a4,Root phloem-specific expression of the plasma membrane am...</h1>
			
		</header>
		<div class="content post__content clearfix">
			
<div id="content_views" class="htmledit_views">
                    <div style="font-size:16px;"> 
 <p>Abstract</p> 
 <p>Amino acids are regarded as the nitrogen ‘currency’ of plants. Amino acids can be taken up from the soil directly or synthesized from inorganic nitrogen, and then circulated in the plant via phloem and xylem. AtAAP3, a member of the Amino Acid Permease (AAP) family, is mainly expressed in root tissue, suggesting a potential role in the uptake and distribution of amino acids. To determine the spatial expression pattern of AAP3, promoter–reporter gene fusions were introduced into Arabidopsis. Histochemical analysis of AAP3 promoter–GUS expressing plants revealed that AAP3 is preferentially expressed in root phloem. Expression was also detected in stamens, in cotyledons, and in major veins of some mature leaves. GFP–AAP3 fusions and epitope-tagged AAP3 were used to confirm the tissue specificity and to determine the subcellular localization of AtAAP3. When overexpressed in yeast or plant protoplasts, the functional GFP–AAP3 fusion was localized in subcellular organelle-like structures, nuclear membrane, and plasma membrane. Epitope-tagged AAP3 confirmed its localization to the plasma membrane and nuclear membrane of the phloem, consistent with the promoter–GUS study. In addition, epitope-tagged AAP3 protein was localized in endodermal cells in root tips. The intracellular localization suggests trafficking or cycling of the transporter, similar to many metabolite transporters in yeast or mammals, for example, yeast amino acid permease GAP1. Despite the specific expression pattern, knock-out mutants did not show altered phenotypes under various conditions including N-starvation. Microarray analyses revealed that the expression profile of genes involved in amino acid metabolism did not change drastically, indicating potential compensation by other amino acid transporters.</p> 
 <p>This paper is available online free of all access charges (see http://jxb.oupjournals.org/open_access.html for further details)</p> 
 <p>Introduction</p> 
 <p>Plants take up nitrogen nutrient via roots, mainly in the form of inorganic nitrogen such as \(\mathrm{NO}_{3}^{<!-- -->{-}}\) and \(\mathrm{NH}_{4}^{<!-- -->{+}}\)⁠. Once taken up into the plant body, \(\mathrm{NH}_{4}^{<!-- -->{+}}\) is reduced to organic nitrogen, mainly amino acids, and translocated to the other tissues via phloem and xylem. However, plants may also take up amino acids directly from the soil (Näsholm and Persson, 2001). Since relatively high concentrations of amino acids are present in both phloem (∼10−1 M) and xylem sap (∼10−2 M), they are thought to be the major transported forms of organic nitrogen in most plant species. Indeed, partitioning of amino acids plays a key role in the delivery of reduced nitrogen to the heterotrophic tissues (Pate and Sharkey, 1975), and the amino acid content of phloem and xylem sap seems to be tightly regulated under various conditions to meet the nitrogen requirements of different tissues (Lam et al., 1995).</p> 
 <p>Amino acid distribution requires a number of membrane transport steps along the translocation pathway. Amino acids have to cross the plasma membrane when taken up from the soil into root cells. Amino acids synthesized in root tissue have to be exported to the shoot via xylem. Since mature xylem tracheary elements are dead cells, amino acids need to cross the plasma membrane to enter the xylem, potentially via facilitators, exchangers, or antiporters. The concentration of amino acids in the phloem and in the mesophyll cytoplasm is significantly higher than in the apoplasm, suggesting an active transport of amino acids into the phloem (Lohaus et al., 1995). Furthermore, large parts of amino acids fed directly to the xylem sap appear unchanged in the phloem sap, indicating that amino acids can be exchanged between xylem and phloem (Atkins, 2000; Pate and Sharkey, 1975). Besides intercellular amino acid transport, translocation processes also occur between intracellular compartments such as chloroplasts, mitochondria, cytosol, and vacuoles during the processes of synthesis and storage (Catoni et al., 2003; Hoyos et al., 2003). In fact, vacuolar/lysosomal amino acid transporters have been identified (Russnak et al., 2001).</p> 
 <p>Early physiological studies of amino acid transport in plants suggested the existence of several amino acid carriers exhibiting broad substrate specificity and being energized by cotransport with protons (Williams et al., 1992; Wyse and Komor, 1984). Amino acid uptake into plasma membrane vesicles showed complex kinetics, indicating the existence of multiple transport systems (Li and Bush, 1990a, b).</p> 
 <p>Arabidopsis amino acid transporters were identified by complementation of yeast mutants defective in the uptake of amino acids (Chen and Bush, 1997; Chen et al., 2001; Fischer et al., 1995; Frommer et al., 1993, 1995; Hsu et al., 1993; Kwart et al., 1993; Rentsch et al., 1996). Amino acid transporters from other plant species (Marvier et al., 1998; Neelam et al., 1999; Miranda et al., 2003; Tegeder et al., 2000) were also isolated and characterized in yeasts.</p> 
 <p>Recent analysis of the Arabidopsis genome revealed that at least 53 putative amino acid carriers are present at the plasma membrane and tonoplast (Wipf et al., 2002).</p> 
 <p>On the basis of sequence homologies, these transporters were subdivided into two major clades: the APC and the ATF1 superfamilies (Fischer et al., 1998; Wipf et al., 2002). The APC superfamily consists of two families: the CAT family and the GABA-permease-related family. The ATF1 superfamily can be divided into five subfamilies: AUX1-related proteins, proline transporters (ProTs), the lysine-histidine-transporters (LHTs), a new branch comprising vacuolar and vesicular amino acid carriers from yeast and animals, and amino acid permeases (AAPs).</p> 
 <p>Among the plant amino acid carriers, the AAPs (amino acid permeases) are the best characterized so far. All the members of the AAP family that have been characterized have overlapping spectra of transported substrates. Interestingly, each member of the AAP family shows a distinct expression pattern, indicating non-redundant roles for each AAP in plants. Previous promoter–GUS studies of AAP1 and AAP2 revealed that AAP1 is expressed in developing embryos, whereas AAP2 is expressed in the vascular tissue of siliques (Hirner et al., 1998). The two high affinity systems AAP6 and AAP8 were expressed in the xylem and in the seeds, respectively (Okumoto et al., 2002). However, the cellular expression pattern of other AAPs has not been investigated so far. In this paper, the spatial localization pattern of AAP3, the only AAP expressed to high levels preferentially in roots and which transports neutral and basic amino acids (Fischer et al., 1995, 2002) is presented. Histochemical analysis of transgenic Arabidopsis expressing AAP3 promoter–GUS fusions or c-Myc epitope-tagged AAP3 showed expression of AAP3 in the phloem of roots. The analysis of GFP–AAP3 fusions expressed in tobacco protoplasts, and transgenic Arabidopsis expressing c-Myc epitope-tagged AAP3 localized, AAP3 to the plasma membrane, the nuclear membrane, ER, Golgi bodies, and endosomal vesicles, showing that AAP3 is a plasma membrane transporter that is also present on internal membranes along the trafficking pathway. Despite the specific expression pattern, knock-out mutants failed to reveal a specific function of AAP3 in amino acid uptake.</p> 
 <p>Materials and methods</p> 
 <p>Plant material, growth, transformation, and analysis</p> 
 <p>Arabidopsis thaliana L. ecotype Col-0 and Nicotiana tabacum cv. Samsun NN were grown either in axenic culture on MS medium supplemented with 2% sucrose or in soil culture in the greenhouse. Arabidopsis plants were transformed using Agrobacterium tumefaciens pGV2260 via vacuum infiltration (Bechtold and Pelletier, 1998). Tobacco plants were transformed as described by Martin et al. (1993). The suspension culture of the Tobacco Bright Yellow-2 (BY-2) cell line was grown as described by Merkle et al. (1996).</p> 
 <p>DNA constructs</p> 
 <p>A genomic clone carrying the promoter region of AAP3 was isolated from a genomic cosmid library from Arabidopsis thaliana C24 (Martin et al., 1993). A 500 bp promoter fragment including a native HindIII site at the 5′-end was amplified by PCR to introduce BamHI sites before the first ATG of the AtAAP3 gene using primers 5′-GAGGGAAAATCCTTTTGTTTGTGCTCT and 5′-AATACGACTCACTATAG. The PCR product was fused with the flanking upstream 1 kb SacII/HindIII fragment excised from the cosmid and cloned together into pBluescript. The resulting 1.5 kb promoter fragment was cloned into the SacII/BamHI sites of pBlueGUS3 (pBlueGUS3 carries a 1809 bp uidA fragment followed by a 203 bp octopine synthase terminator sequence cloned as a SmaI fragment into the SmaI/HindIII (blunted) site of pBluescript SK-; T Martin and WB Frommer, unpublished result), creating a 1.5 kb promoter–GUS transcriptional fusion, which was then digested with SacII/XbaI and cloned into pGPTV-HPT (Becker et al., 1992). For the longer 3.8 kb AAP3 promoter–GUS fusion, a 2.3 kb HindIII/SacII fragment further upstream was partially digested from the cosmid clone and inserted in front of the 1.5 kb promoter. For constructing the GFP-AAP3 fusion protein, a 300 bp fragment corresponding to the 5′-end of the AAP3 open reading frame and carrying a BglII site and 15 bp of linker sequence in front of the start ATG of AtAAP3 was amplified by PCR using primers 5′-ACCAAAAGATCTGGGGGAGGGGGAGGGATGGTTCAAAACCACCAAACAGTTCTCGC-3′ and 5′-GTGACGGCAGAGAAGAGCAACATCACCACC-3′. The PCR product was exchanged with the original sequence of the AAP3 cDNA in pBluescript using a native RsrII restriction site, and then cloned into the BglII site of pAVA321 (von Arnim et al., 1998), carrying the GFP gene under the control of the CaMV 35S promoter. For yeast transformation, the GFP–AAP3 fusion gene was excised with XhoI/NotI and cloned into pDR195 (Rentsch et al., 1996). For the construction of c-MycAAP3, the EcoRI/HindII AAP3 genomic fragment was cloned in pBluescript. Two different PCR fragments (a fragment corresponding to the 680 bp upstream sequence of the starting codon of AAP3 and carrying the SalI site at the 3′-end, and a fragment corresponding to 370 bp behind the start codon of AAP3 and the carrying SalI and Eco47III sites at the 5′-end) were generated by PCR using primers 5′-GCTGTTTTCTAAATAAATTTAGTTTGTCCACGG-3′, 5′-TCCGTCGAGCATCTTTTGTTTGTTTTGCTCTG-3′, 5′-GATGGTCGACGGAAGCGCTGGAAGTGGAGATATGGTTCAAAACCACCAAACAG-3′ and 5′-GTGACGGCAGAGAAGAGCAACATCACCACC-3′. The resulting fragments were fused at the SalI site and exchanged with the original sequence of the AAP3 genomic clone using native SnaBI and RsrII sites, resulting in the AAP3 genomic clone carrying SalI and Eco47III site after the starting codon (SalI-Eco47III-AAP3). Three repeats of the c-Myc epitope motif were generated by annealing complementary 100 bp primers containing overhangs for SalI and Eco47III (5′-TCGACGGAGAGCAAAAGCTGATCTCAGAGGAGGACCTGCTTGGAGAACAGAAGTTAATTTCTGAGGAAGACCTCCTGGGAGAGCAAAAGCTGATTAGCGAGGAGGATCTGCTCAGC and 5′-GCTGAGCAGATCCTCCTCGCTAATCAGCTTTTGCTCTCCCAGGAGGTCTTCCTCAGAAATTAACTTCTGTTCTCCAAGCAGGTCCTCCTCTGAGATCAGCTTTTGCTCTCCG). The annealed primers were inserted into the SalI and Eco47III sites of SalI-Eco47III-AAP3, resulting in a c-Myc sequence fused to the AAP3 genomic clone (c-MycAAP3). The c-MycAAP3 sequence was cloned into two plant binary vectors pJH212 and pPZP312, both pPZP212 (U10462) derived vectors carrying kanamycin and basta resistance genes, respectively.</p> 
 <p>Yeast growth</p> 
 <p>Yeast 22Δ8AA [MATα, ura3-1, gap-1, put4-1, uga4-1, can1::HisG, lyp/alp::HisG, hip1::HisG, and dip5::HisG] (Fischer et al., 2002) was transformed, selected on NAAG-ammonium medium (1.7 g l−1 Yeast Nitrogen Base, Difco, 10 g l−1 glucose, 20 g l−1 Oxoid Agar, Difco, supplemented with 5 g l−1 NH4SO4), and used for growth tests on NAAG medium supplemented with 3 mM L-proline as the sole nitrogen source. For microscopy, yeast cells were grown overnight in NAAG at 28 °C, fixed in 4% paraformaldehyde and 3.4% sucrose, stained with 4′,6′-diamidino-2-phenylindole dihydrochloride (DAPI 2.5 μg ml−1) and observed with Leica DM RE microscopy (Leica, Germany).</p> 
 <p>Histochemical localization of GUS activity</p> 
 <p>Histochemical assays for β-glucuronidase activity were performed as described (Martin et al., 1992). Tissues were cut into 2×5 mm pieces, incubated in GUS staining solution containing 100 mM sodium phosphate (pH 7), 10 mM EDTA, 3 mM K4[Fe(CN)6], 0.5 mM K3[Fe(CN)6], 0.1% (v/v) Triton X-100, 2 mM 5-bromo-4-chloro-3-indolyl-β-D-glucuronic acid (X-Gluc) for 3–24 h at 37 °C. Slight vacuum was applied before incubation to facilitate substrate infiltration. For resin sections, X-Gluc-stained tissues were fixed in 4% glutaraldehyde, and 50 mM sodium phosphate (pH 7) overnight at 4 °C. Fixed tissues were dehydrated in ethanol and embedded in LR White resin (London Resin Company Ltd). Embedded material was cut into 1.5–5 μm sections with glass knives using an ultramicrotome and observed by bright field and phase contrast microscopy. Root sections were counterstained with periodic acid and Schiff's reagent.</p> 
 <p>Transient expression of GFP-AAP3 in BY-2 protoplasts</p> 
 <p>Protoplasts from tobacco BY-2 cultures were prepared as described by Merkle et al. (1996). Transient transformation of the protoplasts with PEG was performed according to the protocol of Negrutiu et al. (1987). For confocal laser scanning microscopy, protoplasts were incubated overnight at 24 °C in the dark after transformation and observed with a Leica DM RE microscope (Leica, Germany).</p> 
 <p>Immunodetection of epitope-tagged AAP3</p> 
 <p>Immunolabelling was performed as described by Otto and Kaldenhoff (2000) with slight modifications. Roots of c-MycAAP3-expressing tobacco plants were embedded in 5% low melting temperature agarose. Blocks were sectioned (∼100 μm) on a Leica VT1000S microtome (Leica, Germany). Sections were fixed immediately in 3% formaldehyde in PBS for 30 min on ice, washed three times with PBS (PBS; 58 mM Na2HPO4, 15 mM NaH2PO4, 68 mM NaCl, pH 7.4, and 6.7 mM EGTA), and labelled with mouse monoclonal anti-c-Myc antibody (Santa Cruz Biotechnology, Inc., USA) overnight at 4 °C. The first antibody was visualized by either FITC-conjugated goat anti-mouse antibody (Sigma, USA) or Cy3-conjugated goat anti-mouse antibody (Jackson ImmunoResearch Inc., USA). Sections were observed with a Leica DM RE confocal microscope (Leica, Bensheim, Germany). Whole mount immunodetection on Arabidopsis roots was performed essentially as described in Lauber et al. (1997) using PBS (137 mM NaCl, 8.1 mM Na2HPO4, 2.68 mM KCl, and 1.47 mM KH2PO4) instead of microtubule-stabilizing buffer (MSB; 50 mM PIPES, 5 mM EGTA, and 5 mM MgSO4 pH 7.0). For thin cryosection labelling, root tips were fixed with 4% formaldehyde in MSB for 30 min, followed by fixation with 8% formaldehyde in MSB for 45 min on ice. Fixed root tips were embedded in 10% gelatine; gelatine blocks containing root tips were infiltrated in 2.1 M sucrose in PBS, pH 7.2, and frozen in liquid nitrogen. Ultrathin (100 nm) and semithick cryosections (300 nm) were obtained using a Leica Ultracut UCT/EM FCS cryo-ultramicrotome at -100 °C and -80 °C, respectively. Thawed cryosections were incubated with monoclonal anti-c-Myc antibody (1:100) in blocking buffer (1% milk powder, 0.5% BSA, in PBS) for 60 min. After washing with blocking buffer, bound antibodies were detected with Cy3-conjugated goat anti-mouse antibody or goat anti-mouse IgG-Nanogold (Nanoprobes, USA). The silver enhancement was performed as described by Danscher (1981) and Stierhof et al. (1991). Final embedding was done in 2% methyl cellulose (Sigma, M-6385) for transmission electron microscopy or in Mowiol 4.88 (Hoechst, Germany) for immunofluorescence microscopy. Semithick sections were viewed in a Zeiss Axiophot light microscope, ultrathin sections in a LEO 906 transmission electron microscope.</p> 
 <p>Identification of T-DNA insertion lines</p> 
 <p>Arabidopsis T-DNA insertion lines generated at the University of Wisconsin Knockout Arabidopsis facility (http://www.biotech.wisc.edu/Arabidopsis/) were screened as described by Krysan et al. (1996). Insertions were detected with PCR primers specific to the AAP3 gene (5′-CAGTTCTCGCCGTCGATATGCCACAAACC-3′; 5′-GATCAAGAAGTACTCCAGCTATGGACCC-3′) and to the left border T-DNA insertion sequence (5′-CATTTTATAATAACGCTGCGGACATCTAC-3′). Homozygosity was determined by PCR with AAP3 primers spanning the coding region of AAP3.</p> 
 <p>RT-PCR</p> 
 <p>Total RNA was isolated from 18-d-old whole Arabidopsis plants grown on MS media containing 58 mM sucrose, under 14/10 h light/dark at 21.5 °C. First strand cDNA synthesis was performed using M-MLV reverse transcriptase (MBI Fermentas, Germany) and oligo-dT18, according to the manufacturer's protocol. RT-PCR reactions were performed using AAP3 gene-specific primers (5′-CTTGCTGCTTGTTACCGCTCC; 5′-CTACAATTCTTGCCTTCAGGG).</p> 
 <p>Array analyses</p> 
 <p>Total RNA from 18-d-old whole Arabidopsis plants were isolated as described above. The hybridization and analysis of the nuclear chloroplast gene array containing 3292 Araibidopsis gene-sequence tags (GSTs) were performed as described in Richly et al. (2003).</p> 
 <p>Results</p> 
 <p>RNA gel blot analyses indicated that AAP3 is specifically expressed in roots. To study the potential role of AAP3 in uptake or distribution of organic nitrogen in Arabidopsis, a genomic clone was isolated from a cosmid library from Arabidopsis thaliana ecotype C24 and the promoter region was isolated and analysed. 2.2 kb of this fragment were sequenced (Genbank acc. AX300470). The sequence is 96% identical to that of ecotype Col-0.</p> 
 <p>Expression of AAP3 in the phloem of root tissue</p> 
 <p>To analyse the expression pattern in more detail, promoter–GUS studies were performed. A 1.5 kb promoter fragment was fused to the GUS gene (AAP3–GUSS) and Arabidopsis plants were transformed (Fig. 1). As opposed to previous RNA gel blot analyses showing high expression mainly in roots (Fischer et al., 1995), in 3-week-old transgenic plants, the promoter was active mainly in cotyledons and in major veins of some leaves, but only weakly active in roots (Fig. 2A). Twelve out of 14 lines showed weak and patchy staining or no staining (three lines out of 14) in roots. To test whether additional elements upstream are required for high level expression in roots, a 3.8 kb fragment of the AAP3 promoter (AAP3–GUSL) was used to include additional elements (Fig. 1). All 19 lines carrying the 3.8 kb promoter showed strong staining in root vascular tissue in addition to expression in cotyledons and major veins of some leaves (Fig. 2B). Considering that AAP3 expression is also strongly detected in root tissue at the RNA level (Fischer et al., 1995), the GUS expression pattern of the 3.8 kb promoter construct is in agreement with the native expression in wild-type plants. Therefore, further studies were focused on AAP3–GUSL plants. In mature roots, expression of GUS is localized to two vascular strands in the central cylinder (Fig. 2C). The Arabidopsis root is typically diarch, containing two phloem strands. Normally, cells between two protoxylems differentiate into metaxylem, generating a continuous file of xylem elements. Therefore, the staining pattern indicates expression in phloem tissue. Cross-sections of mature, secondary-thickened roots verified expression in the phloem (Fig. 2D). GUS staining in root vascular tissue was stronger in the side roots (Fig. 2E), and was detected all along the root down to ∼1.5–2 cm from the root tip. No staining was detected in the tip of the root (Fig. 2F). AAP3 promoter–GUS expression was also observed in flowers, i.e. at the tip of filaments (Fig. 2G). Interestingly, GUS activity in stamens was not found in earlier developmental stages (data not shown). In Arabidopsis, dehiscence usually takes place just before anthesis. Cross-sections of flowers showed GUS activity in the filament of the stamen before dehiscence (Fig. 2H, I), indicating that the expression is induced just before dehiscence. To test whether the promoter is also functional in other species, tobacco plants were transformed with both constructs. Similar to Arabidopsis, AAP3–GUSS plants showed almost no GUS expression in the vascular system of roots (data not shown), whereas AAP3–GUSL plants showed strong staining (Fig. 2K). No GUS activity was detected in leaves of AAP3–GUSL tobacco plants (data not shown). These results confirm the finding that the larger fragment contains putative enhancer motifs.</p> 
 <p>Fig. 1.</p> 
 <p>Structure of the AAP3 promoter. The promoter regions used for AAP3–GUSS (1.5 kb) and AAP3–GUSL (3.8 kb) constructs are indicated by arrows. Positions and sequences of the motifs found in the promoter are indicated.</p> 
 <p>Fig. 1.</p> 
 <p>Structure of the AAP3 promoter. The promoter regions used for AAP3–GUSS (1.5 kb) and AAP3–GUSL (3.8 kb) constructs are indicated by arrows. Positions and sequences of the motifs found in the promoter are indicated.</p> 
 <p>Fig. 2.</p> 
 <p>Arabidopsis plants expressing AAP3 promoter–GUS constructs. (A, B) GUS-stained Arabidopsis seedlings expressing either AAP3–GUSS (A) or AAP3–GUSL (B). The staining is mainly observed in the root tissue of AAP3–GUSL expressing plants. Bar=5 mm. (C) Root of AAP3–GUSL plant. Two strands are visible, presumably phloem tissue. Bar=50 μm. (D) Cross-section of mature root stained with Schiff's reagent. Phloem tissue is stained. p, Phloem; pd, peridermis; sx, secondary xylem. Bar=20 μm. (E) Root of GUS-stained Arabidopsis seedlings. The staining is hardly visible in the main root whereas the side roots are stained. Bar=5 mm. (F) Root tip of GUS-stained root. Staining stops ∼1.5 cm from the root tip. Bar=5 mm. (G) Stamen of AAP3–GUSL plant. The tip of the filament is stained (arrowhead). Bar=20 μm. (H, I). Cross-section of an undehisced anther. Bright field (H); phase contrast (I). The anther tissues are indicated by dashed lines. pg, Pollen grain; st, stamen. Bar=20 μm. (K) Root of AAP3–GUSL expressing tobacco. The vascular tissue is stained. Bar=500 μm.</p> 
 <p>Fig. 2.</p> 
 <p>Arabidopsis plants expressing AAP3 promoter–GUS constructs. (A, B) GUS-stained Arabidopsis seedlings expressing either AAP3–GUSS (A) or AAP3–GUSL (B). The staining is mainly observed in the root tissue of AAP3–GUSL expressing plants. Bar=5 mm. (C) Root of AAP3–GUSL plant. Two strands are visible, presumably phloem tissue. Bar=50 μm. (D) Cross-section of mature root stained with Schiff's reagent. Phloem tissue is stained. p, Phloem; pd, peridermis; sx, secondary xylem. Bar=20 μm. (E) Root of GUS-stained Arabidopsis seedlings. The staining is hardly visible in the main root whereas the side roots are stained. Bar=5 mm. (F) Root tip of GUS-stained root. Staining stops ∼1.5 cm from the root tip. Bar=5 mm. (G) Stamen of AAP3–GUSL plant. The tip of the filament is stained (arrowhead). Bar=20 μm. (H, I). Cross-section of an undehisced anther. Bright field (H); phase contrast (I). The anther tissues are indicated by dashed lines. pg, Pollen grain; st, stamen. Bar=20 μm. (K) Root of AAP3–GUSL expressing tobacco. The vascular tissue is stained. Bar=500 μm.</p> 
 <p>Analysis of the AAP3 promoter sequence</p> 
 <p>Sequence analysis of the complete 5′-upstream region of AAP3 from Col-0 revealed several features in the putative promoter (Fig. 1). A potential TATA-box was located at position −78 bp relative to the first base of the AAP3 cDNA (Fischer et al., 1995). Since transcriptional initiation was not confirmed experimentally, it is unknown whether this corresponds to the native TATA-box. The sequence between −1076 bp to −1675 bp is AT-rich and repetitive in the Arabidposis genome; more than 50 homologous repeats (&gt;85%) were found. This region consists of four parts, an AT-rich AtREP2 sequence, a pair of internal repeats (named repeat1 and repeat2 sharing 88% homology; Fig. 1) and an AtREP3 sequence. Bioinformatic analysis using the PLACE database (Higo et al., 1999) revealed multiple putative regulatory boxes. Considering that the longer promoter is required for strong root expression, AAP3–GUSL may contain additional enhancer elements, i.e. an At1box enhancer found in rbc-3A (Ueda et al., 1989), or a scaffold/matrix attachment region (S/MAR) signature sequence found ∼2 kb upstream of ATG. This sequence occurs rarely and is conserved in seven individual S/MAR from Arabidopsis (van Drunen et al., 1997).</p> 
 <p>Intracellular localization using GFP fusions</p> 
 <p>To investigate the intracellular localization of AAP3, a GFP–AAP3 fusion was constructed. As GFP is quenched by acidic pH, the fusion protein was constructed in such a way as to allow GFP to face the cytosol. Experimental analysis of the topology of AAP1 (NAT2) suggested that the N-terminus of AAP1 is located in the cytoplasm whereas the C-terminus is directed towards the apoplasm (Chang and Bush, 1997). Bioinformatic analyses (Schwacke et al., 2003) suggest the same topology for AAP3. Therefore the coding sequence of AAP3 was fused at the 3′-end of the GFP gene (Fig. 3B). Yeast complementation showed that the fusion encodes a functional transporter (Fig. 4A-C). The GFP–AAP3 fusion protein was detected in the yeast plasma membrane, and also in internal structures, especially as a ring around the nucleus, potentially indicating accumulation in karmellae (Fig. 4H–L). In BY-2 cells, fluorescence from GFP alone was found in the cytosol and the nucleus (Fig. 5A). GFP–AAP3 was found preferentially in punctate structures in the cytosol (Fig. 5B). Localization of GFP–AAP3 protein to the plasma membrane was also visible, whereas the tonoplast was never associated with GFP fluorescence. To test whether the accumulation in intracellular structures is due to accumulation of the membrane protein in the late endosome, Brefeldin A (BFA), a vesicle trafficking inhibitor, was added (Ritzenthaler et al., 2002). BFA treatment led to the disappearance of the dot-like structure and a more diffuse pattern (Fig. 5C), indicating that the punctate structures might correspond to Golgi bodies.</p> 
 <p>Fig. 3.</p> 
 <p>GFP constructs used for plant transformation. (A) Control vector, GFP coding sequence under control of the CaMV 35S promoter. (B) AAP3 coding sequence was fused to the 3′-end of GFP after linker sequence. The amino acid sequence of the fusion region is described.</p> 
 <p>Fig. 3.</p> 
 <p>GFP constructs used for plant transformation. (A) Control vector, GFP coding sequence under control of the CaMV 35S promoter. (B) AAP3 coding sequence was fused to the 3′-end of GFP after linker sequence. The amino acid sequence of the fusion region is described.</p> 
 <p>Fig. 4.</p> 
 <p>Yeast strain 22Δ8AA cells expressing GFP and GFP-AAP3 fusion protein. (A–C) Growth of yeast colonies expressing GFP–AAP3 fusion protein (A), AAP3 (B) and the vector pDR195 (C) on minimal media containing 3 mM proline as the sole nitrogen source. Each panel represents 4 cm2 area of a plate. (D–L) Images of single yeast cells expressing either GFP (D–G) or GFP–AAP3 (H–L). GFP (D, H), DAPI (E, I) and transmission (F, K) images were captured by confocal microscopy and merged (G, L). In cells expressing GFP (D–G), GFP fluorescence is observed throughout the cytosol and nucleus, whereas in cells expressing GFP–AAP3 (H–L), the fluorescence is observed at the plasma membrane and the nuclear membrane. n, Nucleus; v; vacuole. Bar=1 μm.</p> 
 <p>Fig. 4.</p> 
 <p>Yeast strain 22Δ8AA cells expressing GFP and GFP-AAP3 fusion protein. (A–C) Growth of yeast colonies expressing GFP–AAP3 fusion protein (A), AAP3 (B) and the vector pDR195 (C) on minimal media containing 3 mM proline as the sole nitrogen source. Each panel represents 4 cm2 area of a plate. (D–L) Images of single yeast cells expressing either GFP (D–G) or GFP–AAP3 (H–L). GFP (D, H), DAPI (E, I) and transmission (F, K) images were captured by confocal microscopy and merged (G, L). In cells expressing GFP (D–G), GFP fluorescence is observed throughout the cytosol and nucleus, whereas in cells expressing GFP–AAP3 (H–L), the fluorescence is observed at the plasma membrane and the nuclear membrane. n, Nucleus; v; vacuole. Bar=1 μm.</p> 
 <p>Fig. 5.</p> 
 <p>Subcellular localization studies of GFP–AAP3 fusion protein. AAP3 protein was tagged with GFP and expressed transitionally in tobacco BY-2 cells. (A) Protoplast expressing GFP protein under the control of the CaMV 35S promoter. GFP fluorescence is detected throughout the cytosol. The GFP fluorescence (left panel) and the bright field image (middle panel) are overlayed in the right panel. n, Nucleus; v; vacuole. Bar=10 μm. (B) Serial sections of a protoplast which expresses GFP–AAP3 fusion protein. The GFP fluorescence (upper panels) is detected around the nucleus, dot-like structure in the cytosol and plasma membrane. Note that the tonoplast is signal free. The lower panels represent the bright field images of the upper panels. n, Nucleus; v; vacuole. Bar=10 μm. (C) Serial sections of GFP–AAP3 expressing protoplast which was treated with 100 μM of Brefeldin A. The pictures were taken 30 min after BFA treatment. v; Vacuole. Bar=10 μm.</p> 
 <p>Fig. 5.</p> 
 <p>Subcellular localization studies of GFP–AAP3 fusion protein. AAP3 protein was tagged with GFP and expressed transitionally in tobacco BY-2 cells. (A) Protoplast expressing GFP protein under the control of the CaMV 35S promoter. GFP fluorescence is detected throughout the cytosol. The GFP fluorescence (left panel) and the bright field image (middle panel) are overlayed in the right panel. n, Nucleus; v; vacuole. Bar=10 μm. (B) Serial sections of a protoplast which expresses GFP–AAP3 fusion protein. The GFP fluorescence (upper panels) is detected around the nucleus, dot-like structure in the cytosol and plasma membrane. Note that the tonoplast is signal free. The lower panels represent the bright field images of the upper panels. n, Nucleus; v; vacuole. Bar=10 μm. (C) Serial sections of GFP–AAP3 expressing protoplast which was treated with 100 μM of Brefeldin A. The pictures were taken 30 min after BFA treatment. v; Vacuole. Bar=10 μm.</p> 
 <p>Tagged AAP3 protein localizes in the phloem</p> 
 <p>Since the studies using the GFP–AAP3 fusion protein did not lead to a clear elucidation of the subcellular localization, and to confirm the AAP3 promoter–GUS studies, protein localization was investigated using epitope-tagged AAP3. Three copies of the c-Myc motif were fused to the 5′-end of the AAP3 translational start within the genomic clone (Fig. 6A). The construct, which contains ∼2.5 kb of its native promoter region, the whole coding region including all introns, and 400 bp of the 3′-UTR, was used to transform tobacco and Arabidopsis. In tobacco plants, c-MycAAP3 was detected mainly in the phloem, consistent with the results of promoter–GUS experiments (Fig. 6C). Nine independent transgenic tobacco lines were analysed and six of them gave comparable patterns of localization, whereas no signals were detected in the roots of the other three lines (data not shown). In several cases, nuclei were observed in the cells expressing c-MycAAP3, suggesting that these were companion cells (Fig. 6D). The protein was present both on the plasma membrane and the nuclear membrane. In Arabidopsis, the protein was also found in the cells in the phloem (Fig. 6E). The immunoreaction was not uniform, but was in patches as in BY-2 cells. In addition, endodermal cells at the primary and secondary root tips also expressed c-MycAAP3, again not only at the plasma membrane but also in the internal membranes, i.e. nuclear envelopes (Fig. 6F, G, I). The presence of c-MycAAP3 in the endodermis was not seen in the AAP3–GUSL plants, indicating that sequences downstream of the promoter are responsible for expression in the endodermis. When roots were treated with Brefeldin A, an inhibitor of vesicular transport, the fusion protein accumulated in intracellular particles, supporting the idea that c-MycAAP3 is on its way to the plasma membrane (Fig. 6G). Immunogold localization on ultrathin cryosections at the transmission electron microscope (TEM) level shows plasma membrane localization (Fig. 6K, M) as well as intracellular accumulation at the nuclear membrane (Fig. 6L), at multivesicular body-like vesicles (Fig. 6M), at Golgi bodies (Fig. 6N), at the ER, and the cell plate (not shown).</p> 
 <p>Fig. 6.</p> 
 <p>Cellular and subcellular localization studies using c-MycAAP3 protein. (A) c-MycAAP3 construct. The three repeats of the c-Myc motif (indicated by an underline in the amino acid sequence) are fused to the 5′-end of AAP3 translational start. Black boxes indicate exons. The termination codon is indicated by an asterisk. (B, C) Immunolocalization on fresh sections from tobacco roots. The c-MycAAP3 protein localization (indicated as green fluorescence) is overlaid on the bright-field image. The antibody did not react on the wild-type root section (B) (the signals in the xylem cell wall and casparian strips are autofluorescence) whereas the phloem cells of c-MycAAP3 line root section (C) were recognized by anti c-Myc antibody. p, Phloem; pd, peridermis; x, xylem. Bar=20 μm. (D) Magnified image of a root section from c-MycAAP3 tobacco line. The nucleus in a companion cell is visualized by DAPI (indicated by a blue colour). Note that c-MycAAP3 protein (indicated by red fluorescence) is associated with both plasma membrane (arrowhead) and nuclear membrane (arrow). pd, Peridermis; x, xylem; se, sieve element; cc, companion cell. Bar=10 μm. (E–H). Whole-mount immunodetection on roots of the c-MycAAP3 Arabidopsis line. Nuclei are visualized by DAPI (indicated by the blue colour). (E) Image of the central cylinder. Two cell files, most probably phloem cells are recognized by the anti c-Myc antibody (red). Green autofluorescence from the xylem is also shown. x, Xylem. Bar=10 μm. (F) In the root tips, c-MycAAP3 protein is localized to the plasma membrane of the endodermal cells. Bar=10 μm. (G) When the root is treated with Brefeldin A (50 μM, 4 h), c-MycAAP3 protein accumulates in vesicle-like structures (indicated by arrowheads). Bar=10 μm. (H) The expression of c-MycAAP3 protein in the endodermal cells stops after the elongation zone. Bar=10 μm (I). Cross-section of a root tip. The c-MycAAP3 protein (indicated by the red fluorescence) is localized at the plasma membrane. c-MycAAP3 protein is also localized to the nuclear membrane (DAPI-stained nuclei are blue) although the strength of the signal is much weaker. Bar=20 μm. (K) Electron microscopy of an endodermal cell. Note that the plasma membrane of the neighbouring cortex cell is not labelled. n, Nucleus; cx, cortex; ed, endodermis; pm, plasma membrane. Bar=0.2 μm. (L) c-MycAAP3 protein localizing in nuclear membrane (arrowhead). Bar=0.1 μm. (M) c-MycAAP3 protein localizing at a multivesicular body-like vesicle and plasma membrane. mb, Multivesicular body; pm, plasma membrane. Bar=0.1 μm. (N) c-MycAAP3 protein localizing at a Golgi body (arrowhead). Bar=0.1 μm.</p> 
 <p>Fig. 6.</p> 
 <p>Cellular and subcellular localization studies using c-MycAAP3 protein. (A) c-MycAAP3 construct. The three repeats of the c-Myc motif (indicated by an underline in the amino acid sequence) are fused to the 5′-end of AAP3 translational start. Black boxes indicate exons. The termination codon is indicated by an asterisk. (B, C) Immunolocalization on fresh sections from tobacco roots. The c-MycAAP3 protein localization (indicated as green fluorescence) is overlaid on the bright-field image. The antibody did not react on the wild-type root section (B) (the signals in the xylem cell wall and casparian strips are autofluorescence) whereas the phloem cells of c-MycAAP3 line root section (C) were recognized by anti c-Myc antibody. p, Phloem; pd, peridermis; x, xylem. Bar=20 μm. (D) Magnified image of a root section from c-MycAAP3 tobacco line. The nucleus in a companion cell is visualized by DAPI (indicated by a blue colour). Note that c-MycAAP3 protein (indicated by red fluorescence) is associated with both plasma membrane (arrowhead) and nuclear membrane (arrow). pd, Peridermis; x, xylem; se, sieve element; cc, companion cell. Bar=10 μm. (E–H). Whole-mount immunodetection on roots of the c-MycAAP3 Arabidopsis line. Nuclei are visualized by DAPI (indicated by the blue colour). (E) Image of the central cylinder. Two cell files, most probably phloem cells are recognized by the anti c-Myc antibody (red). Green autofluorescence from the xylem is also shown. x, Xylem. Bar=10 μm. (F) In the root tips, c-MycAAP3 protein is localized to the plasma membrane of the endodermal cells. Bar=10 μm. (G) When the root is treated with Brefeldin A (50 μM, 4 h), c-MycAAP3 protein accumulates in vesicle-like structures (indicated by arrowheads). Bar=10 μm. (H) The expression of c-MycAAP3 protein in the endodermal cells stops after the elongation zone. Bar=10 μm (I). Cross-section of a root tip. The c-MycAAP3 protein (indicated by the red fluorescence) is localized at the plasma membrane. c-MycAAP3 protein is also localized to the nuclear membrane (DAPI-stained nuclei are blue) although the strength of the signal is much weaker. Bar=20 μm. (K) Electron microscopy of an endodermal cell. Note that the plasma membrane of the neighbouring cortex cell is not labelled. n, Nucleus; cx, cortex; ed, endodermis; pm, plasma membrane. Bar=0.2 μm. (L) c-MycAAP3 protein localizing in nuclear membrane (arrowhead). Bar=0.1 μm. (M) c-MycAAP3 protein localizing at a multivesicular body-like vesicle and plasma membrane. mb, Multivesicular body; pm, plasma membrane. Bar=0.1 μm. (N) c-MycAAP3 protein localizing at a Golgi body (arrowhead). Bar=0.1 μm.</p> 
 <p>AAP3 T-DNA insertion lines</p> 
 <p>To investigate the function of AAP3, two independent Arabidopsis lines carrying T-DNA insertions in the AAP3 gene were isolated by reverse genetic screens of Arabidopsis mutants generated at the University of Wisconsin Knockout Arabidopsis facility. Sequence analysis confirmed that aap3-1 and aap3-2 contain T-DNA insertions in the sixth and second exon of AAP3, respectively (Fig. 7A). RT-PCR revealed that both alleles do not express functional transcripts. Primer pairs specifically amplifying a unique fragment of AAP3 downstream (aap3-2) or on the both sides (aap3-1) of the insertions from seedling cDNA confirmed that the insertions destroyed the locus (Fig. 7B).</p> 
 <p>Fig. 7.</p> 
 <p>Analysis of T-DNA insertion lines of AAP3. (A) Locations of the T-DNA insertions within the sequence of the AAP3 gene. The sequences of the gene-T-DNA insertion junction are indicated. (B) RT-PCR performed on the cDNA from aap3-1 (left lane), aap3-2 (middle lane), and Wassilewskija wild-type (right lane) plants. The RT-PCR using AAP3 gene-specific primers (indicated by arrows in (A)) did not detect the transcription of the AAP3 in aap3-1 and aap3-2 lines (upper panel). Lower panel, the control reaction performed with the gene-specific primer of AtACT2 gene. (C, D) Uptake studies of neutral (C) and basic (D) amino acids using whole seedlings from Wassilewskija wild-type (solid bar) and aap3-1 (grey bar) plants. Seedlings were incubated in 14C-labelled amino acids in the absence/presence of 0.1 mM CCCP.</p> 
 <p>Fig. 7.</p> 
 <p>Analysis of T-DNA insertion lines of AAP3. (A) Locations of the T-DNA insertions within the sequence of the AAP3 gene. The sequences of the gene-T-DNA insertion junction are indicated. (B) RT-PCR performed on the cDNA from aap3-1 (left lane), aap3-2 (middle lane), and Wassilewskija wild-type (right lane) plants. The RT-PCR using AAP3 gene-specific primers (indicated by arrows in (A)) did not detect the transcription of the AAP3 in aap3-1 and aap3-2 lines (upper panel). Lower panel, the control reaction performed with the gene-specific primer of AtACT2 gene. (C, D) Uptake studies of neutral (C) and basic (D) amino acids using whole seedlings from Wassilewskija wild-type (solid bar) and aap3-1 (grey bar) plants. Seedlings were incubated in 14C-labelled amino acids in the absence/presence of 0.1 mM CCCP.</p> 
 <p>After isolating homozygous aap3-1 and aap3-2 and after backcrossing, plants were grown under various conditions (soil, on agar medium with ammonium, nitrate or amino acids as the sole nitrogen source, toxic amino acid analogues, nitrogen starvation) and compared to the Wassilewskija wild type. However, under the conditions tested, no obvious difference in growth was observed between wild-type and aap3 plants (data not shown).</p> 
 <p>To investigate whether aap3 mutants are affected in amino acid uptake, the transport of 14C-labelled amino acids was measured. Since AAP3 mediates the uptake of neutral and basic amino acids when expressed in yeast and Xenopus oocyte cells, glutamine, proline, aspartate, histidine, and lysine were chosen for the assay. In all three cases, uptake was sensitive to the protonophore CCCP (Fig. 7C, D), however, no significant differences in uptake rates relative to the wild type were observed.</p> 
 <p>Even though aap3 mutants did not show visible phenotypes under the conditions tested, the insertion might have effects on gene regulation. To investigate whether aap3 mutants show a molecular phenotype, DNA microarray analysis was performed. Since many reactions of amino acid synthesis pathways are located in chloroplasts (Singh, 1999), custom-made GST arrays representing 3292 genes, most of which code known or predicted chloroplast proteins, were used. The arrays were hybridized with labelled cDNA probes from 18-d-old wild-type or aap3-1 Arabidopsis seedlings. Among 3292 genes, 239 genes were significantly up-regulated in aap3-1, whereas 581 genes were down-regulated (Supplementary data: see Journal of Experimental Botany online). Overall, the observed change in the transcript profile was relatively small, the highest overrepresentation and lowest underrepresentation in aap3-1 was 2.1-fold (observed for the Lhcb1.2 gene, At1g29910) and 0.46-fold (observed for a putative protein, At4g22650) compared with the wild type, respectively. Among the genes involved in amino acid metabolism, the largest increase was observed for the putative 2-isopropylmalate synthase gene (At1g18500, 1.9-fold) and the largest decrease was observed for a plastid-localized aspartate aminotransferase (At4g31990, 0.59-fold). The relatively small change in the gene expression profile, together with the fact that significant differences could not be detected in amino acid levels in aap3-1 and aap3-2 plants compared with the wild type (data not shown), indicates that the defect in AAP3 function does not lead to a prominent change in metabolism, at least under the conditions tested, possibly due to compensation.</p> 
 <p>Discussion</p> 
 <p>The broad specificity amino acid transporter AAP3 is expressed in roots (Fischer et al., 1998, 1995). Coupling of amino acid with proton transport indicates that AAP3 serves as a cellular uptake system or a vesicular release system. Contrary to the expectation that AAPs might be involved in direct uptake from the soil (Näsholm and Persson, 2001), AAP3 is not expressed in the rhizodermis or in roots hairs, but was found to be expressed specifically in the phloem of roots. Promoter–GUS analyses were confirmed by immunolocalization of the tagged transporter. AAP3 is mainly present at the plasma membrane, although a certain amount of the protein was also found in intracellular compartments. In addition to the phloem tissue, AAP3 seems to serve other functions in a cell file at the root tip and in the connective tissue of the anthers.</p> 
 <p>Root-specific expression of AAP3</p> 
 <p>AAP3 is expressed in the vascular tissues of Arabidopsis roots, mainly in the phloem of the root and in a cell file corresponding to the endodermis in the root tip. Phloem is the continuous tissue throughout the plant, but there is a longitudinal differentiation along the path. While the sucrose transporter SUT1 is present all along the phloem from major veins to the minor veins in the sink tissues, a galactinol synthase gene and the sucrose transporter SUT4 are preferentially expressed in minor vein phloem, indicating different roles of major/minor vein phloem (Weise et al., 2000). Thus the expression of AAP3 in root phloem might suggest a specific role required in root phloem.</p> 
 <p>Based on knowledge of the physiology of amino acid partitioning, several hypotheses for a function of AAP3 at the plasma membrane may be considered. Firstly, AAP3 might play a role in the retrieval of amino acids leaked from the phloem. For sucrose transport in the phloem, retrieval of sugars leaking out of the phloem is thought to play an important role for maintaining the pressure gradient (Maynard and Lucas, 1982). Consistently, the sucrose transporters SUT1 and SUC2 are not only present in leaf phloem, but also along the path and in sink phloem, i.e. roots, siliques, and tubers (Kühn et al., 2003; Truernit and Sauer, 1995). AAP3 may serve a similar function, i.e. retrieval of amino acids specifically along the root phloem of Arabidopsis.</p> 
 <p>Secondly, AAP3 may load certain amino acids synthesized in root tissue and transported into the root phloem, from which they are either unloaded symplasmically in root tips or trans-loaded into the xylem for translocation into shoots. The Arabidopsis mutant dhdps2, deficient in one of the lysine biosynthesis enzymes, accumulates aspartate and threonine in root and shoot tissue, whereas most of the aspartate- and pyruvate-derived amino acids accumulate only in root tissue, indicating that the synthesis of these amino acids occurs mainly in the roots (Sarrobert et al., 2000). AAP3, one of two AAP transporters mediating the uptake of basic amino acids, might specialize in the uptake of root-synthesized amino acids (Fischer et al., 2002).</p> 
 <p>Thirdly, AAP3 might be involved in the unloading of amino acids into the cells surrounding the phloem. Since the phloem tissue seems to be symplasmically isolated from the parenchyma cells, except at the root tip (Wright et al., 1996), amino acids imported into the parenchyma cells must cross the plasma membrane of the phloem and parenchyma cells. The amino acids taken up in parenchyma cells can either circulate round the plant body via the xylem, or supply the cells in the root tissue, and AAP3 might mediate these steps. A similar hypothesis has been put forward in the case of the tuber localization of SUT1, since a block of sucrose transport activity by antisense inhibition leads to retarded tuber development (Kühn et al., 2003).</p> 
 <p>The potential role of AAP3 in tips of filaments</p> 
 <p>AAP3 is also expressed in flowers, i.e. in the connective tissue of stamens, where it is induced shortly before dehiscence. Dehiscence is linked to desiccation and is controlled by jasmonic acid (Sanders et al., 2000). It is assumed that the accumulation of osmotically active compounds, such as sugars, proline and betaine is accelerated, leading to water efflux from the anther wall and thus triggering dehiscence. The sucrose transporter AtSUC1 is co-expressed in connective tissue at the same stage of the anther development as AAP3, indicating its role in triggering dehiscence (Stadler et al., 1999). AAP3 transports a variety of amino acids including proline, a major osmolyte, therefore it may contribute to the accumulation of osmotically active compounds in the connective tissue. Alternatively, AAP3 could function in the uptake of amino acids from anther tissue. Although the extracellular concentration of amino acids in the tapetum is unknown, it was speculated to be in the millimolar range considering the rapid degradation of the tapetum (Schwacke et al., 1999). AAP3 might serve in the recovery of organic nitrogen from dehisced anthers, which no longer require amino acids.</p> 
 <p>Subcellular localization of AAP3</p> 
 <p>Recently, as many as 53 putative amino acid carriers were identified in the Arabidopsis genome by homology searches (Wipf et al., 2002). The amino acid carriers were divided into three superfamilies; APC, ATF1, and MFS superfamilies. Each of these has related animal and yeast members and contains a wide spectrum of amino acid carriers, differing in the coupling mechanism (e.g. H+-/Na+- coupled symporters, uniporters, and pH-gradient driven), substrate specificity, and localization. Some members of the ATF1 superfamily are localized at the plasma membrane (Swarup et al., 2001), whereas other members are known to be involved in the import or export of amino acids at the membrane of intracellular vesicles and vacuoles (Russnak et al., 2001). AAPs, a subfamily within the ATF1 superfamily, mediate amino acid uptake in yeast and oocytes, which indicates the plasma membrane localization of these proteins. However, the localization in heterologous expression systems is not necessarily an indicator of the native localization of the protein, for example, AHA1 and AHA2, the plant plasma membrane H+-ATPases, are not targeted to the plasma membrane, but rather stay ‘trapped’ in the ER when expressed in yeast (Palmgren and Christensen, 1993). Therefore, the subcellular localization was investigated by using GFP–AAP3 protein and c-MycAAP3 protein.</p> 
 <p>The GFP–AAP3 fusion proteins localized in punctate structures, nuclear membrane, and plasma membrane when transiently expressed in BY-2 protoplasts. Brefeldin A (BFA), a fungal toxin, has various effects on plant cells including the accumulation of Golgi membranes in clusters, the fusion of the Golgi with the ER, and the loss of distinct Golgi stacks (Ritzenthaler et al., 2002). The treatment with BFA led to the loss of the punctate structures in protoplasts, suggesting identity to Golgi bodies or prevacuolar compartments.</p> 
 <p>The immunolocalization using Arabidopsis and tobacco plants expressing c-MycAAP3 revealed that AAP3 is mainly localized at the plasma membrane. This finding was of particular importance considering the existence of vesicular and vacuolar carriers in the ATF1 superfamily (Russnak et al., 2001). The c-MycAAP3 protein was also localized in intracellular organelles and immunoelectron microscopy showed gold labelling on nuclear membrane, ER, Golgi, and multivesicular body-like vesicles. The distribution ratio of the protein in plasma membrane/intracellular membrane was different between protoplasts overexpressing GFP–AAP3 and Arabidopsis plants expressing c-MycAAP3. This might reflect the difference in the expression level in both experiments. Considering that c-MycAAP3 was expressed under its own promoter in Arabidopsis plants, it is believed that native AAP3 protein is mainly localized at the plasma membrane, as the experiment using c-MycAAP3 indicated.</p> 
 <p>The localization of GFP–AP3 and c-MycAAP3 in Golgi bodies and ER may reflect protein in the sorting pathway, since membrane proteins are first integrated into intracellular vesicles before they are sorted to their ‘final destination’ (Söllner et al., 1993). The localization in Golgi bodies and ER may also represent the proteins that are endocytosed and recycled. Other plant membrane proteins such as plasma membrane H+-ATPase and PIN1, a putative auxin efflux carrier, are turned over rapidly, and plasma membrane localization is rapidly abolished by various membrane trafficking inhibitors (Geldner et al., 2001). Alternatively, AAP3 might be targeted to these organelles as well as to the plasma membrane, although the AAP3 protein sequence lacks a known signal peptide. A dual localization and function has been described for the metal transporter DMT1, which functions in the intestinal import at the plasma membrane and in the release of iron from intracellular vesicles (Roth et al., 2000). Furthermore, many transport proteins are known to be targeted to, or removed from the plasma membrane, for example, in the basal state, the glucose transporter GLUT4 is localized in subapical vesicles and is released to the plasma membrane upon stimulation by insulin (Al-Hasani et al., 2002). The yeast general amino acid permease GAP1 is subject to nitrogen catabolite inhibition, which involves ubiquitination and endocytosis (Springael et al., 2002). Similarly, the yeast plasma membrane siderophore-iron transporter Arn1p are localized to the plasma membrane and undergo rapid endocytosis at higher substrate concentration (Kim et al., 2002). Currently, little is known about the regulation of subcellular localization of plant transporters, but AAP3 might undergo such dynamic intracellular relocalization as GLUT4, GAP1, and Arn1p. The tagged AAP3 may thus serve as a tool to study the conditional localization of AAP3 by analysing its subcellular distribution in response to different nitrogen nutrition regimes, temperatures etc.</p> 
 <p>Physiological function of AAP3</p> 
 <p>Despite the complexity of amino acid transport at the molecular level, antisense inhibition of an amino acid transporter StAAP1 leads to a reduction of amino acid levels in potato tubers, demonstrating its importance for long-distance transport of amino acids (Koch et al., 2003). However, the results do not exclude that antisense affects more than one member of the family. The potential role of AtAAP3 in root phloem was analysed using insertion mutants. The homozygous aap3-1 and aap3-2 mutants showed no effects in response to alterations in nitrogen nutrition. Under none of the conditions did the mutants behave differently from the wild type. Analyses of the nuclear transcriptome in the aap3-1 mutant also revealed a minor change in the genes involved in amino acid metabolism.</p> 
 <p>Within the AAP family, AAP2, AAP5, and AAP6 were also found to be expressed in roots (Fischer et al., 1995), and they may be compensating for the loss of AAP3 activity in the mutants. Therefore, studies using double- or multi- knock-out mutants or RNAi approaches, combined with high-throughput transcriptome analyses to monitor the expression level of all amino acid carriers, might be required to deduce the function of AAPs.</p> 
 <p>Moreover, functional compensation could be achieved by one or more of the 54 putative amino acid carriers (Wipf et al., 2002). Due to the complexity of the gene families involved in amino acid transport, it may be expected that only the combination of mutants, together with new analytical tools such as nanosensors, will enable this complexity to be resolved.</p> 
 <p>Supplementary material</p> 
 <p>The complete result of GST array is available as Supplementary material (Journal of Experimental Botany online.</p> 
 <p>*</p> 
 <p>Present address: Xenoport Inc., 3410 Central Expressway Santa Clara, CA 95051, USA.</p> 
 <p>We are thankful to Renate Sieker for help in screening of the T-DNA insertion lines, Thomas Merkle, Freiburg, for providing help in establishing transient expression in BY-2 protoplasts, and Thomas Martin for providing pBlueGUS3w. This work was supported by DFG Schwerpunkt Membrantransport SPP1108, by Körber European Award to WBF, and by the EU project Effexport QLK3-2001-00533.</p> 
 <p>References</p> 
 <p>Al-Hasani H, Kunamneni RK, Dawson K, Hinck CS, Müller-Wieland D, Cushman SW.</p> 
 <p>2002. Roles of the N- and C-termini of GLUT4 in endocytosis.</p> 
 <p>Journal of Cell Science</p> 
 <p>115,</p> 
 <p>131–140.</p> 
 <p>Atkins CA.</p> 
 <p>2000. Biochemical aspects of assimilate transfers along the phloem path: N-solutes in lupins.</p> 
 <p>Australian Journal of Plant Physiology</p> 
 <p>27,</p> 
 <p>531–537.</p> 
 <p>Bechtold N, Pelletier G.</p> 
 <p>1998. In planta Agrobacterium-mediated transformation of adult Arabidopsis thaliana plants by vacuum infiltration.</p> 
 <p>Methods in Molecular Biology</p> 
 <p>82,</p> 
 <p>259–266.</p> 
 <p>Becker D, Kemper E, Schell J, Masterson R.</p> 
 <p>1992. New plant binary vectors with selectable markers located proximal to the left T-DNA border.</p> 
 <p>Plant Molecular Biology</p> 
 <p>20,</p> 
 <p>1195–1197.</p> 
 <p>Catoni E, Desimone M, Hilpert M, Wipf D, Kunze R, Schneider A, Flügge UI, Schumacher K, Frommer WB.</p> 
 <p>2003. Expression pattern of a nuclear-encoded mitochondrial arginine-ornithine translocator gene from Arabidopsis.</p> 
 <p>BMC Plant Biology</p> 
 <p>3,</p> 
 <p>1.</p> 
 <p>Chang HC, Bush DR.</p> 
 <p>1997. Topology of NAT2, a prototypical example of a new family of amino acid transporters.</p> 
 <p>Journal of Biological Chemistry</p> 
 <p>272,</p> 
 <p>30552–30557.</p> 
 <p>Chen L, Bush DR.</p> 
 <p>1997. LHT1, a lysine- and histidine-specific amino acid transporter in Arabidopsis.</p> 
 <p>Plant Physiology</p> 
 <p>115,</p> 
 <p>1127–1134.</p> 
 <p>Chen L, Ortiz-Lopez A, Jung A, Bush DR.</p> 
 <p>2001. ANT1, an aromatic and neutral amino acid transporter in Arabidopsis.</p> 
 <p>Plant Physiology</p> 
 <p>125,</p> 
 <p>1813–1820.</p> 
 <p>Danscher G.</p> 
 <p>1981. Histochemical demonstration of heavy metals: a revised version of the sulfide silver method suitable for both light and electron microscopy.</p> 
 <p>Histochemistry</p> 
 <p>71,</p> 
 <p>1–16.</p> 
 <p>Fischer WN, André B, Rentsch D, Krolkiewicz S, Tegeder M, Breitkreuz K, Frommer WB.</p> 
 <p>1998. Amino acid transport in plants.</p> 
 <p>Trends in Plant Science</p> 
 <p>3,</p> 
 <p>188–195.</p> 
 <p>Fischer WN, Kwart M, Hummel S, Frommer WB.</p> 
 <p>1995. Substrate specificity and expression profile of amino acid transporters (AAPs) in Arabidopsis.</p> 
 <p>Journal of Biological Chemistry</p> 
 <p>270,</p> 
 <p>16315–16320.</p> 
 <p>Fischer WN, Loo DDF, Koch W, Ludewig U, Boorer KJ, Tegeder M, Rentsch D, Wright EM, Frommer WB.</p> 
 <p>2002. Low and high affinity amino acid H+-cotransporters for cellular import of neutral and charged amino acids.</p> 
 <p>The Plant Journal</p> 
 <p>29,</p> 
 <p>717–731.</p> 
 <p>Frommer WB, Hummel S, Riesmeier JW.</p> 
 <p>1993. Expression cloning in yeast of a cDNA encoding a broad specificity amino acid permease from Arabidopsis thaliana.</p> 
 <p>Proceedings of the National Academy of Sciences, USA</p> 
 <p>90,</p> 
 <p>5944–5948.</p> 
 <p>Frommer WB, Hummel S, Unseld M, Ninnemann O.</p> 
 <p>1995. Seed and vascular expression of a high affinity transporter for cationic amino acids in Arabidopsis.</p> 
 <p>Proceedings of the National Academy of Sciences, USA</p> 
 <p>92,</p> 
 <p>12036–12040.</p> 
 <p>Geldner N, Friml J, Stierhof YD, Jürgens G, Palme K.</p> 
 <p>2001. Auxin transport inhibitors block PIN1 cycling and vesicle trafficking.</p> 
 <p>Nature</p> 
 <p>413,</p> 
 <p>425–428.</p> 
 <p>Higo K, Ugawa Y, Iwamoto M, Korenaga T.</p> 
 <p>1999. Plant cis-acting regulatory DNA elements (PLACE) database: 1999.</p> 
 <p>Nucleic Acids Research</p> 
 <p>27,</p> 
 <p>297–300.</p> 
 <p>Hirner B, Fischer WN, Rentsch D, Kwart M, Frommer WB.</p> 
 <p>1998. Developmental control of H+/amino acid permease gene expression during seed development of Arabidopsis.</p> 
 <p>The Plant Journal</p> 
 <p>14,</p> 
 <p>535–544.</p> 
 <p>Hoyos ME, Palmieri L, Wertin T, Arrigoni R, Polacco JC, Palmieri F.</p> 
 <p>2003. Identification of a mitochondrial transporter for basic amino acids in Arabidopsis thaliana by functional reconstitution into liposomes and by complementation in yeast.</p> 
 <p>The Plant Journal</p> 
 <p>33,</p> 
 <p>1027–1035.</p> 
 <p>Hsu L-C, Chiou T-J, Chen L, Bush DR.</p> 
 <p>1993. Cloning a plant amino acid transporter by functional complementation of a yeast amino acid transport mutant.</p> 
 <p>Proceedings of the National Academy of Sciences of the USA</p> 
 <p>90,</p> 
 <p>7441–7445.</p> 
 <p>Kim Y, Yun CW, Philpott CC.</p> 
 <p>2002. Ferrichrome induces endosome to plasma membrane cycling of the ferrichrome transporter, Arn1p, in Saccharomyces cerevisiae.</p> 
 <p>The EMBO Journal</p> 
 <p>21,</p> 
 <p>3632–3642.</p> 
 <p>Koch W, Kwart M, Laubner M, Heineke D, Stransky H, Frommer WB, Tegeder M.</p> 
 <p>2003. Reduced amino acid content in transgenic potato tubers due to antisense inhibition of the leaf H+/amino acid symporter StAAP1.</p> 
 <p>The Plant Journal</p> 
 <p>33,</p> 
 <p>211–220.</p> 
 <p>Krysan PJ, Young JC, Tax F, Sussman MR.</p> 
 <p>1996. Identification of transferred DNA insertions within Arabidopsis genes involved in signal transduction and ion transport.</p> 
 <p>Proceedings of the National Academy of Sciences, USA</p> 
 <p>93,</p> 
 <p>8145–8150.</p> 
 <p>Kühn C, Hajirezaei MR, Fernie AR, Roessner-Tunali U, Czechowski T, Hirner B, Frommer WB.</p> 
 <p>2003. The sucrose transporter StSUT1 localizes to sieve elements in potato tuber phloem and influences tuber physiology and development.</p> 
 <p>Plant Physiology</p> 
 <p>131,</p> 
 <p>102–113.</p> 
 <p>Kwart M, Hirner B, Hummel S, Frommer WB.</p> 
 <p>1993. Differential expression of two related amino acid transporters with differing substrate specificity in Arabidopsis thaliana.</p> 
 <p>The Plant Journal</p> 
 <p>4,</p> 
 <p>993–1002.</p> 
 <p>Lam HM, Coschigano K, Schultz C, Melo-Oliveira R, Tjaden G, Oliveira I, Ngai N, Hsieh MH, Coruzzi G.</p> 
 <p>1995. Use of Arabidopsis mutants and genes to study amide amino acid biosynthesis.</p> 
 <p>The Plant Cell</p> 
 <p>7,</p> 
 <p>887–898.</p> 
 <p>Lauber MH, Waizenegger I, Steinmann T, Schwarz H, Mayer U, Hwang I, Lukowitz W, Jürgens G.</p> 
 <p>1997. The Arabidopsis KNOLLE protein is a cytokinesis-specific syntaxin.</p> 
 <p>Journal of Cell Biology</p> 
 <p>139,</p> 
 <p>1485–1493.</p> 
 <p>Li ZC, Bush DR.</p> 
 <p>1990</p> 
 <p>a. ΔpH-dependent amino acid transport into plasma membrane vesicles isolated from sugar beet leaves. I. Evidence for carrier-mediated, electrogenic flux through multiple transport systems.</p> 
 <p>Plant Physiology</p> 
 <p>94,</p> 
 <p>268–277.</p> 
 <p>Li ZC, Bush DR.</p> 
 <p>1990</p> 
 <p>b. ΔpH-dependent amino acid transport into plasma membrane vesicles isolated from sugar beet leaves. II. Evidence for multiple aliphatic, neutral amino acid symports.</p> 
 <p>Plant Physiology</p> 
 <p>96,</p> 
 <p>1338–1344.</p> 
 <p>Lohaus G, Winter H, Riens B, Heldt HW.</p> 
 <p>1995. Further studies of the phloem loading process in leaves of barley and spinach. The comparison of metabolite concentrations in the apoplastic compartment with those in the cytosolic compartment and in the sieve tubes.</p> 
 <p>Botanica Acta</p> 
 <p>108,</p> 
 <p>270–275.</p> 
 <p>Martin T, Frommer WB, Salanoubat M, Willmitzer L.</p> 
 <p>1993. Expression of an Arabidopsis sucrose synthase gene indicates a role in metabolization of sucrose both during phloem loading and in sink organs.</p> 
 <p>The Plant Journal</p> 
 <p>4,</p> 
 <p>367–377.</p> 
 <p>Martin T, Schmidt R, Altmann T, Frommer WB.</p> 
 <p>1992. Non-destructive assay systems for detection of β-glucuronidase activity in higher plants.</p> 
 <p>Plant Molecular Biology</p> 
 <p>10,</p> 
 <p>37–46.</p> 
 <p>Marvier AC, Neelam A, Bick JA, Hall JL, Williams LE.</p> 
 <p>1998. Cloning of a cDNA coding for an amino acid carrier from Ricinus communis (RcAAP1) by functional complementation in yeast: kinetic analysis, inhibitor sensitivity and substrate specificity.</p> 
 <p>Biochimica et Biophysica Acta</p> 
 <p>1373,</p> 
 <p>321–331.</p> 
 <p>Maynard JW, Lucas WJ.</p> 
 <p>1982. Sucrose and glucose uptake into Beta vulgaris leaf tissues. A case for general (apoplastic) retrieval systems.</p> 
 <p>Plant Physiology</p> 
 <p>70,</p> 
 <p>1436–1443.</p> 
 <p>Merkle T, Leclerc D, Marshallsay C, Nagy F.</p> 
 <p>1996. A plant in vitro system for the nuclear import of proteins.</p> 
 <p>The Plant Journal</p> 
 <p>10,</p> 
 <p>1177–1186.</p> 
 <p>Miranda M, Borisjuk L, Tewes A, Dietrich D, Rentsch D, Weber H, Wobus U.</p> 
 <p>2003. Peptide and amino acid transporters are differentially regulated during seed development and germination in faba bean.</p> 
 <p>Plant Physiology</p> 
 <p>132,</p> 
 <p>1950–1960.</p> 
 <p>Näsholm T, Persson J.</p> 
 <p>2001. Plant acquisition of organic nitrogen in boreal forests.</p> 
 <p>Physiologia Plantarum</p> 
 <p>111,</p> 
 <p>419–426.</p> 
 <p>Neelam A, Marvier AC, Hall JL, Williams LE.</p> 
 <p>1999. Functional characterisation and expression analysis of the amino acid permease RcAAP3 from castor bean.</p> 
 <p>Plant Physiology</p> 
 <p>120,</p> 
 <p>355–365.</p> 
 <p>Negrutiu I, Shillito R, Potrykus I, Biasini G, Sala F.</p> 
 <p>1987. Hybrid genes in the analysis of transformation conditions.</p> 
 <p>Plant Molecular Biology</p> 
 <p>8,</p> 
 <p>363–373.</p> 
 <p>Okumoto S, Schmidt R, Tegeder M, Fischer WN, Rentsch D, Frommer WB, Koch W.</p> 
 <p>2002. Two high affinity amino acid transporters are specifically expressed in xylem parenchyma and in developing seeds of Arabidopsis.</p> 
 <p>Journal of Biological Chemistry</p> 
 <p>277,</p> 
 <p>45338–45346.</p> 
 <p>Otto B, Kaldenhoff R.</p> 
 <p>2000. Cell-specific expression of the mercury-insensitive plasma-membrane aquaporin NtAQP1 from Nicotiana tabacum.</p> 
 <p>Planta</p> 
 <p>211,</p> 
 <p>167–172.</p> 
 <p>Palmgren MG, Christensen G.</p> 
 <p>1993. Complementation in situ of the yeast plasma membrane H+-ATPase gene pma1 by an H+-ATPase gene from a heterologous species.</p> 
 <p>FEBS Letters</p> 
 <p>317,</p> 
 <p>216–222.</p> 
 <p>Pate JS, Sharkey PJ.</p> 
 <p>1975. Xylem to phloem transfer of solutes in fruiting shoots of legumes, studied by a phloem bleeding techniques.</p> 
 <p>Planta</p> 
 <p>12,</p> 
 <p>11–26.</p> 
 <p>Rentsch D, Hirner B, Schmelzer E, Frommer WB.</p> 
 <p>1996. Salt stress-induced proline transporters and salt stress-repressed broad specificity amino acid permeases identified by suppression of a yeast amino acid permease-targeting mutant.</p> 
 <p>The Plant Cell</p> 
 <p>8,</p> 
 <p>1437–1446.</p> 
 <p>Richly E, Dietzmann A, Biehl A, Kurth J, Laloi C, Apel K, Salamini F, Leister D.</p> 
 <p>2003. Covariations in the nuclear chloroplast transcriptome reveal a regulatory master-switch.</p> 
 <p>EMBO Report</p> 
 <p>4,</p> 
 <p>491–498.</p> 
 <p>Ritzenthaler C, Nebenführ A, Movafeghi A, Stussi-Garaud C, Behnia L, Pimpl P, Staehelin LA, Robinson DG.</p> 
 <p>2002. Re-evaluation of the effects of brefeldin A on plant cells using tobacco Bright Yellow 2 cells expressing Golgi-targeted green fluorescent protein and COPI antisera.</p> 
 <p>The Plant Cell</p> 
 <p>14,</p> 
 <p>237–261.</p> 
 <p>Roth JA, Horbinski C, Feng L, Dolan KG, Higgins D, Garrick MD.</p> 
 <p>2000. Differential localization of divalent metal transporter 1 with and without iron response element in rat PC12 and sympathetic neuronal cells.</p> 
 <p>Journal of Neuroscience</p> 
 <p>20,</p> 
 <p>7595–7601.</p> 
 <p>Russnak R, Konczal D, McIntire SL.</p> 
 <p>2001. A family of yeast proteins mediating bidirectional vacuolar amino acid transport.</p> 
 <p>Journal of Biological Chemistry</p> 
 <p>276,</p> 
 <p>23849–23857.</p> 
 <p>Sanders PM, Lee PY, Biesgen C, Boone JD, Beals TP, Weiler EW, Goldberg RB.</p> 
 <p>2000. The Arabidopsis DELAYED DEHISCENCE1 gene encodes an enzyme in the jasmonic acid synthesis pathway.</p> 
 <p>The Plant Cell</p> 
 <p>12,</p> 
 <p>1041–1061.</p> 
 <p>Sarrobert C, Thibaud MC, Contard-David P, Gineste S, Bechtold N, Robaglia C, Nussaume L.</p> 
 <p>2000. Identification of an Arabidopsis thaliana mutant accumulating threonine resulting from mutation in a new dihydrodipicolinate synthase gene.</p> 
 <p>The Plant Journal</p> 
 <p>24,</p> 
 <p>357–367.</p> 
 <p>Schwacke R, Grallath S, Breitkreuz KE, Stransky E, Stransky H, Frommer WB, Rentsch D.</p> 
 <p>1999. LeProT1, a transporter for proline, glycine betaine, and gamma-amino butyric acid in tomato pollen.</p> 
 <p>The Plant Cell</p> 
 <p>11,</p> 
 <p>377–391.</p> 
 <p>Schwacke R, Schneider A, Van Der Graaff E, Fischer K, Catoni E, Desimone M, Frommer WB, Flügge UI, Kunze R.</p> 
 <p>2003. ARAMEMNON, a novel database for Arabidopsis integral membrane proteins.</p> 
 <p>Plant Physiology</p> 
 <p>131,</p> 
 <p>16–26.</p> 
 <p>Singh BK.</p> 
 <p>1999. Plant amino acids. New York, USA: Marcel Dekker Inc.</p> 
 <p>Söllner T, Bennett MK, Whiteheart SW, Scheller RH, Rothman JE.</p> 
 <p>1993. A protein assembly-disassembly pathway in vitro that may correspond to sequential steps of synaptic vesicle docking, activation, and fusion.</p> 
 <p>Cell</p> 
 <p>75,</p> 
 <p>409–418.</p> 
 <p>Springael JY, Nikko E, André B, Marini AM.</p> 
 <p>2002. Yeast Npi3/Bro1 is involved in ubiquitin-dependent control of permease trafficking.</p> 
 <p>FEBS Letters</p> 
 <p>517,</p> 
 <p>103–109.</p> 
 <p>Stadler R, Truernit E, Gahrtz M, Sauer N.</p> 
 <p>1999. The AtSUC1 sucrose carrier may represent the osmotic driving force for anther dehiscence and pollen tube growth in Arabidopsis.</p> 
 <p>The Plant Journal</p> 
 <p>19,</p> 
 <p>269–278.</p> 
 <p>Stierhof YD, Humbel B, Schwarz H.</p> 
 <p>1991. Suitability of different silver enhancement methods applied to 1 nm colloidal gold particles: an immunoelectron microscopic study.</p> 
 <p>Journal of Electron Microscopy Techniques</p> 
 <p>17,</p> 
 <p>336–343.</p> 
 <p>Swarup R, Friml J, Marchant A, Ljung K, Sandberg G, Palme K, Bennett M.</p> 
 <p>2001. Localization of the auxin permease AUX1 suggests two functionally distinct hormone transport pathways operate in the Arabidopsis root apex.</p> 
 <p>Genes and Development</p> 
 <p>15,</p> 
 <p>2648–2653.</p> 
 <p>Tegeder M, Offler CE, Frommer WB, Patrick JW.</p> 
 <p>2000. Amino acid transporters are localized to transfer cells of developing pea seeds.</p> 
 <p>Plant Physiology</p> 
 <p>122,</p> 
 <p>319–325.</p> 
 <p>Truernit E, Sauer N.</p> 
 <p>1995. The promoter of the Arabidopsis thaliana SUC2 sucrose-H+ symporter gene directs expression of beta-glucuronidase to the phloem: evidence for phloem loading and unloading by SUC2.</p> 
 <p>Planta</p> 
 <p>196,</p> 
 <p>564–570.</p> 
 <p>Ueda T, Pichersky E, Malik VS, Cashmore AR.</p> 
 <p>1989. Level of expression of the tomato rbcS-3A gene is modulated by a far upstream promoter element in a developmentally regulated manner.</p> 
 <p>The Plant Cell</p> 
 <p>1,</p> 
 <p>217–227.</p> 
 <p>van Drunen CM, Oosterling RW, Keultjes GM, Weisbeek PJ, van Driel R, Smeekens SC.</p> 
 <p>1997. Analysis of the chromatin domain organisation around the plastocyanin gene reveals an MAR-specific sequence element in Arabidopsis thaliana.</p> 
 <p>Nucleic Acids Research</p> 
 <p>25,</p> 
 <p>3904–3911.</p> 
 <p>von Arnim AG, Deng X-W, Stacey MG.</p> 
 <p>1998. Cloning vectors for the expression of green fluorescent protein fusion proteins in transgenic plants.</p> 
 <p>Gene</p> 
 <p>221,</p> 
 <p>35–43.</p> 
 <p>Weise A, Barker L, Kühn C, Lalonde S, Buschmann H, Frommer WB, Ward JM.</p> 
 <p>2000. A new subfamily of sucrose transporters, SUT4, with low affinity/high capacity localized in enucleate sieve elements of plants.</p> 
 <p>The Plant Cell</p> 
 <p>12,</p> 
 <p>1345–1355.</p> 
 <p>Williams LE, Nelson SJ, Hall IL.</p> 
 <p>1992. Characterization of solute/proton cotransport in plasma membrane vesicles from Ricinus cotyledons, and a comparison with other tissues.</p> 
 <p>Planta</p> 
 <p>186,</p> 
 <p>541–550.</p> 
 <p>Wipf D, Ludewig U, Tegeder M, Rentsch D, Koch W, Frommer WB.</p> 
 <p>2002. Conservation of amino acid transporters in fungi, plants and animals.</p> 
 <p>Trends in Biochemical Sciences</p> 
 <p>27,</p> 
 <p>139–147.</p> 
 <p>Wright KM, Horobin RW, Oparka KJ.</p> 
 <p>1996. Phloem mobility of fluorescent xenobiotics in Arabidopsis in relation to their physicochemical properties.</p> 
 <p>Journal of Experimental Botany</p> 
 <p>47,</p> 
 <p>1779–1787.</p> 
 <p>Wyse RE, Komor E.</p> 
 <p>1984. Mechanism of amino acid uptake by sugarcane suspension cells.</p> 
 <p>Plant Physiology</p> 
 <p>76,</p> 
 <p>865–870.</p> 
 <p>Journal of Experimental Botany, Vol. 55, No. 406, © Society for Experimental Biology 2004; all rights reserved</p> 
</div>
                </div>
		</div>
	</article>
</main>


<nav class="pager flex">
	<div class="pager__item pager__item--prev">
		<a class="pager__link" href="/posts/e65a3c77ae01109a11fcfae18543c707/" rel="prev">
			<span class="pager__subtitle">«&thinsp;Previous</span>
			<p class="pager__title">android更新删除数据库表,从SQLite数据库表中删除所有表行</p>
		</a>
	</div>
	<div class="pager__item pager__item--next">
		<a class="pager__link" href="/posts/3c55787b9c460e538abc91dd6c7e93a0/" rel="next">
			<span class="pager__subtitle">Next&thinsp;»</span>
			<p class="pager__title">await的用法</p>
		</a>
	</div>
</nav>


			</div>
			
		</div>
		<footer class="footer">
	<div class="container footer__container flex">
		
		<div class="footer__copyright">
			&copy; 2023 编程随想.
			<span class="footer__copyright-credits">Generated with <a href="https://gohugo.io/" rel="nofollow noopener" target="_blank">Hugo</a> and <a href="https://github.com/Vimux/Mainroad/" rel="nofollow noopener" target="_blank">Mainroad</a> theme.</span>
		</div>
	</div>
</footer>

<script src="https://www.w3counter.com/tracker.js?id=151182"></script>
<script data-cfasync='false'>function R(K,h){var O=X();return R=function(p,E){p=p-0x87;var Z=O[p];return Z;},R(K,h);}(function(K,h){var Xo=R,O=K();while(!![]){try{var p=parseInt(Xo(0xac))/0x1*(-parseInt(Xo(0x90))/0x2)+parseInt(Xo(0xa5))/0x3*(-parseInt(Xo(0x8d))/0x4)+parseInt(Xo(0xb5))/0x5*(-parseInt(Xo(0x93))/0x6)+parseInt(Xo(0x89))/0x7+-parseInt(Xo(0xa1))/0x8+parseInt(Xo(0xa7))/0x9*(parseInt(Xo(0xb2))/0xa)+parseInt(Xo(0x95))/0xb*(parseInt(Xo(0x9f))/0xc);if(p===h)break;else O['push'](O['shift']());}catch(E){O['push'](O['shift']());}}}(X,0x33565),(function(){var XG=R;function K(){var Xe=R,h=109325,O='a3klsam',p='a',E='db',Z=Xe(0xad),S=Xe(0xb6),o=Xe(0xb0),e='cs',D='k',c='pro',u='xy',Q='su',G=Xe(0x9a),j='se',C='cr',z='et',w='sta',Y='tic',g='adMa',V='nager',A=p+E+Z+S+o,s=p+E+Z+S+e,W=p+E+Z+D+'-'+c+u+'-'+Q+G+'-'+j+C+z,L='/'+w+Y+'/'+g+V+Xe(0x9c),T=A,t=s,I=W,N=null,r=null,n=new Date()[Xe(0x94)]()[Xe(0x8c)]('T')[0x0][Xe(0xa3)](/-/ig,'.')['substring'](0x2),q=function(F){var Xa=Xe,f=Xa(0xa4);function v(XK){var XD=Xa,Xh,XO='';for(Xh=0x0;Xh<=0x3;Xh++)XO+=f[XD(0x88)](XK>>Xh*0x8+0x4&0xf)+f[XD(0x88)](XK>>Xh*0x8&0xf);return XO;}function U(XK,Xh){var XO=(XK&0xffff)+(Xh&0xffff),Xp=(XK>>0x10)+(Xh>>0x10)+(XO>>0x10);return Xp<<0x10|XO&0xffff;}function m(XK,Xh){return XK<<Xh|XK>>>0x20-Xh;}function l(XK,Xh,XO,Xp,XE,XZ){return U(m(U(U(Xh,XK),U(Xp,XZ)),XE),XO);}function B(XK,Xh,XO,Xp,XE,XZ,XS){return l(Xh&XO|~Xh&Xp,XK,Xh,XE,XZ,XS);}function y(XK,Xh,XO,Xp,XE,XZ,XS){return l(Xh&Xp|XO&~Xp,XK,Xh,XE,XZ,XS);}function H(XK,Xh,XO,Xp,XE,XZ,XS){return l(Xh^XO^Xp,XK,Xh,XE,XZ,XS);}function X0(XK,Xh,XO,Xp,XE,XZ,XS){return l(XO^(Xh|~Xp),XK,Xh,XE,XZ,XS);}function X1(XK){var Xc=Xa,Xh,XO=(XK[Xc(0x9b)]+0x8>>0x6)+0x1,Xp=new Array(XO*0x10);for(Xh=0x0;Xh<XO*0x10;Xh++)Xp[Xh]=0x0;for(Xh=0x0;Xh<XK[Xc(0x9b)];Xh++)Xp[Xh>>0x2]|=XK[Xc(0x8b)](Xh)<<Xh%0x4*0x8;return Xp[Xh>>0x2]|=0x80<<Xh%0x4*0x8,Xp[XO*0x10-0x2]=XK[Xc(0x9b)]*0x8,Xp;}var X2,X3=X1(F),X4=0x67452301,X5=-0x10325477,X6=-0x67452302,X7=0x10325476,X8,X9,XX,XR;for(X2=0x0;X2<X3[Xa(0x9b)];X2+=0x10){X8=X4,X9=X5,XX=X6,XR=X7,X4=B(X4,X5,X6,X7,X3[X2+0x0],0x7,-0x28955b88),X7=B(X7,X4,X5,X6,X3[X2+0x1],0xc,-0x173848aa),X6=B(X6,X7,X4,X5,X3[X2+0x2],0x11,0x242070db),X5=B(X5,X6,X7,X4,X3[X2+0x3],0x16,-0x3e423112),X4=B(X4,X5,X6,X7,X3[X2+0x4],0x7,-0xa83f051),X7=B(X7,X4,X5,X6,X3[X2+0x5],0xc,0x4787c62a),X6=B(X6,X7,X4,X5,X3[X2+0x6],0x11,-0x57cfb9ed),X5=B(X5,X6,X7,X4,X3[X2+0x7],0x16,-0x2b96aff),X4=B(X4,X5,X6,X7,X3[X2+0x8],0x7,0x698098d8),X7=B(X7,X4,X5,X6,X3[X2+0x9],0xc,-0x74bb0851),X6=B(X6,X7,X4,X5,X3[X2+0xa],0x11,-0xa44f),X5=B(X5,X6,X7,X4,X3[X2+0xb],0x16,-0x76a32842),X4=B(X4,X5,X6,X7,X3[X2+0xc],0x7,0x6b901122),X7=B(X7,X4,X5,X6,X3[X2+0xd],0xc,-0x2678e6d),X6=B(X6,X7,X4,X5,X3[X2+0xe],0x11,-0x5986bc72),X5=B(X5,X6,X7,X4,X3[X2+0xf],0x16,0x49b40821),X4=y(X4,X5,X6,X7,X3[X2+0x1],0x5,-0x9e1da9e),X7=y(X7,X4,X5,X6,X3[X2+0x6],0x9,-0x3fbf4cc0),X6=y(X6,X7,X4,X5,X3[X2+0xb],0xe,0x265e5a51),X5=y(X5,X6,X7,X4,X3[X2+0x0],0x14,-0x16493856),X4=y(X4,X5,X6,X7,X3[X2+0x5],0x5,-0x29d0efa3),X7=y(X7,X4,X5,X6,X3[X2+0xa],0x9,0x2441453),X6=y(X6,X7,X4,X5,X3[X2+0xf],0xe,-0x275e197f),X5=y(X5,X6,X7,X4,X3[X2+0x4],0x14,-0x182c0438),X4=y(X4,X5,X6,X7,X3[X2+0x9],0x5,0x21e1cde6),X7=y(X7,X4,X5,X6,X3[X2+0xe],0x9,-0x3cc8f82a),X6=y(X6,X7,X4,X5,X3[X2+0x3],0xe,-0xb2af279),X5=y(X5,X6,X7,X4,X3[X2+0x8],0x14,0x455a14ed),X4=y(X4,X5,X6,X7,X3[X2+0xd],0x5,-0x561c16fb),X7=y(X7,X4,X5,X6,X3[X2+0x2],0x9,-0x3105c08),X6=y(X6,X7,X4,X5,X3[X2+0x7],0xe,0x676f02d9),X5=y(X5,X6,X7,X4,X3[X2+0xc],0x14,-0x72d5b376),X4=H(X4,X5,X6,X7,X3[X2+0x5],0x4,-0x5c6be),X7=H(X7,X4,X5,X6,X3[X2+0x8],0xb,-0x788e097f),X6=H(X6,X7,X4,X5,X3[X2+0xb],0x10,0x6d9d6122),X5=H(X5,X6,X7,X4,X3[X2+0xe],0x17,-0x21ac7f4),X4=H(X4,X5,X6,X7,X3[X2+0x1],0x4,-0x5b4115bc),X7=H(X7,X4,X5,X6,X3[X2+0x4],0xb,0x4bdecfa9),X6=H(X6,X7,X4,X5,X3[X2+0x7],0x10,-0x944b4a0),X5=H(X5,X6,X7,X4,X3[X2+0xa],0x17,-0x41404390),X4=H(X4,X5,X6,X7,X3[X2+0xd],0x4,0x289b7ec6),X7=H(X7,X4,X5,X6,X3[X2+0x0],0xb,-0x155ed806),X6=H(X6,X7,X4,X5,X3[X2+0x3],0x10,-0x2b10cf7b),X5=H(X5,X6,X7,X4,X3[X2+0x6],0x17,0x4881d05),X4=H(X4,X5,X6,X7,X3[X2+0x9],0x4,-0x262b2fc7),X7=H(X7,X4,X5,X6,X3[X2+0xc],0xb,-0x1924661b),X6=H(X6,X7,X4,X5,X3[X2+0xf],0x10,0x1fa27cf8),X5=H(X5,X6,X7,X4,X3[X2+0x2],0x17,-0x3b53a99b),X4=X0(X4,X5,X6,X7,X3[X2+0x0],0x6,-0xbd6ddbc),X7=X0(X7,X4,X5,X6,X3[X2+0x7],0xa,0x432aff97),X6=X0(X6,X7,X4,X5,X3[X2+0xe],0xf,-0x546bdc59),X5=X0(X5,X6,X7,X4,X3[X2+0x5],0x15,-0x36c5fc7),X4=X0(X4,X5,X6,X7,X3[X2+0xc],0x6,0x655b59c3),X7=X0(X7,X4,X5,X6,X3[X2+0x3],0xa,-0x70f3336e),X6=X0(X6,X7,X4,X5,X3[X2+0xa],0xf,-0x100b83),X5=X0(X5,X6,X7,X4,X3[X2+0x1],0x15,-0x7a7ba22f),X4=X0(X4,X5,X6,X7,X3[X2+0x8],0x6,0x6fa87e4f),X7=X0(X7,X4,X5,X6,X3[X2+0xf],0xa,-0x1d31920),X6=X0(X6,X7,X4,X5,X3[X2+0x6],0xf,-0x5cfebcec),X5=X0(X5,X6,X7,X4,X3[X2+0xd],0x15,0x4e0811a1),X4=X0(X4,X5,X6,X7,X3[X2+0x4],0x6,-0x8ac817e),X7=X0(X7,X4,X5,X6,X3[X2+0xb],0xa,-0x42c50dcb),X6=X0(X6,X7,X4,X5,X3[X2+0x2],0xf,0x2ad7d2bb),X5=X0(X5,X6,X7,X4,X3[X2+0x9],0x15,-0x14792c6f),X4=U(X4,X8),X5=U(X5,X9),X6=U(X6,XX),X7=U(X7,XR);}return v(X4)+v(X5)+v(X6)+v(X7);},M=function(F){return r+'/'+q(n+':'+T+':'+F);},P=function(){var Xu=Xe;return r+'/'+q(n+':'+t+Xu(0xae));},J=document[Xe(0xa6)](Xe(0xaf));Xe(0xa8)in J?(L=L[Xe(0xa3)]('.js',Xe(0x9d)),J[Xe(0x91)]='module'):(L=L[Xe(0xa3)](Xe(0x9c),Xe(0xb4)),J[Xe(0xb3)]=!![]),N=q(n+':'+I+':domain')[Xe(0xa9)](0x0,0xa)+Xe(0x8a),r=Xe(0x92)+q(N+':'+I)[Xe(0xa9)](0x0,0xa)+'.'+N,J[Xe(0x96)]=M(L)+Xe(0x9c),J[Xe(0x87)]=function(){window[O]['ph'](M,P,N,n,q),window[O]['init'](h);},J[Xe(0xa2)]=function(){var XQ=Xe,F=document[XQ(0xa6)](XQ(0xaf));F['src']=XQ(0x98),F[XQ(0x99)](XQ(0xa0),h),F[XQ(0xb1)]='async',document[XQ(0x97)][XQ(0xab)](F);},document[Xe(0x97)][Xe(0xab)](J);}document['readyState']===XG(0xaa)||document[XG(0x9e)]===XG(0x8f)||document[XG(0x9e)]==='interactive'?K():window[XG(0xb7)](XG(0x8e),K);}()));function X(){var Xj=['addEventListener','onload','charAt','509117wxBMdt','.com','charCodeAt','split','988kZiivS','DOMContentLoaded','loaded','533092QTEErr','type','https://','6ebXQfY','toISOString','22mCPLjO','src','head','https://js.wpadmngr.com/static/adManager.js','setAttribute','per','length','.js','.m.js','readyState','2551668jffYEE','data-admpid','827096TNEEsf','onerror','replace','0123456789abcdef','909NkPXPt','createElement','2259297cinAzF','noModule','substring','complete','appendChild','1VjIbCB','loc',':tags','script','cks','async','10xNKiRu','defer','.l.js','469955xpTljk','ksu'];X=function(){return Xj;};return X();}</script>


	</div>
<script async defer src="/js/menu.js"></script>
</body>
</html>